<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 5 Functional Enrichment | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</title>
  <meta name="description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 5 Functional Enrichment | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="github-repo" content="https://github.com/Menna-Arafat/CRC_project" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 5 Functional Enrichment | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  
  <meta name="twitter:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  

<meta name="author" content="Menna Arafat" />


<meta name="date" content="2025-08-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="downstream-visualizations.html"/>
<link rel="next" href="wgcna-pipeline.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network-Based Metabolomics Analysis in R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html"><i class="fa fa-check"></i><b>1</b> Metabolomic Raw data Processing</a>
<ul>
<li class="chapter" data-level="1.1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#internal-functions"><i class="fa fa-check"></i><b>1.1</b> <strong>Internal Functions</strong></a></li>
<li class="chapter" data-level="1.2" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#wrapper-function"><i class="fa fa-check"></i><b>1.2</b> <strong>Wrapper Function</strong></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html"><i class="fa fa-check"></i><b>2</b> Dimensionality Reduction Techniques</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#input-data-preparation-for-metaboanalyst"><i class="fa fa-check"></i><b>2.1</b> <strong>Input data Preparation for Metaboanalyst</strong></a></li>
<li class="chapter" data-level="2.2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pca"><i class="fa fa-check"></i><b>2.2</b> <strong>PCA</strong></a></li>
<li class="chapter" data-level="2.3" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pls-da"><i class="fa fa-check"></i><b>2.3</b> <strong>PLS-DA</strong></a></li>
<li class="chapter" data-level="2.4" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#splsda"><i class="fa fa-check"></i><b>2.4</b> <strong>splsda</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html"><i class="fa fa-check"></i><b>3</b> Metaboanalyst Pipeline</a>
<ul>
<li class="chapter" data-level="3.1" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#normality-assessment"><i class="fa fa-check"></i><b>3.1</b> <strong>Normality Assessment</strong></a></li>
<li class="chapter" data-level="3.2" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#metaboanalystr-object-initialization"><i class="fa fa-check"></i><b>3.2</b> <strong>MetaboAnalystR Object Initialization</strong></a></li>
<li class="chapter" data-level="3.3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#auto-scaleing-z-score-normalization"><i class="fa fa-check"></i><b>3.3</b> <strong>Auto-scaleing (z-score Normalization)</strong></a></li>
<li class="chapter" data-level="3.4" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#post-normalization-normality-check"><i class="fa fa-check"></i><b>3.4</b> <strong>Post-Normalization Normality Check</strong></a></li>
<li class="chapter" data-level="3.5" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.5</b> <strong>Fold change Analysis</strong></a></li>
<li class="chapter" data-level="3.6" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.6</b> <strong>Differential Expression Analysis</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html"><i class="fa fa-check"></i><b>4</b> Downstream Visualizations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#volcano-plot"><i class="fa fa-check"></i><b>4.1</b> <strong>Volcano Plot</strong></a></li>
<li class="chapter" data-level="4.2" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#heatmap-plot"><i class="fa fa-check"></i><b>4.2</b> <strong>Heatmap Plot</strong></a></li>
<li class="chapter" data-level="4.3" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#box-plots"><i class="fa fa-check"></i><b>4.3</b> <strong>Box Plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment.html"><a href="functional-enrichment.html"><i class="fa fa-check"></i><b>5</b> Functional Enrichment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment.html"><a href="functional-enrichment.html#data-preparation"><i class="fa fa-check"></i><b>5.1</b> <strong>Data Preparation</strong></a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment.html"><a href="functional-enrichment.html#enrichment-plot"><i class="fa fa-check"></i><b>5.2</b> <strong>Enrichment plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html"><i class="fa fa-check"></i><b>6</b> WGCNA Pipeline</a>
<ul>
<li class="chapter" data-level="6.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-pipeline-1"><i class="fa fa-check"></i><b>6.1</b> <strong>WGCNA Pipeline</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#choose-appropriate-power-for-soft-thresholding"><i class="fa fa-check"></i><b>6.1.1</b> <strong>Choose appropriate power for soft-thresholding</strong></a></li>
<li class="chapter" data-level="6.1.2" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#set-main-parameters"><i class="fa fa-check"></i><b>6.1.2</b> <strong>Set Main Parameters</strong></a></li>
<li class="chapter" data-level="6.1.3" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-main-function"><i class="fa fa-check"></i><b>6.1.3</b> <strong>WGCNA Main Function</strong></a></li>
<li class="chapter" data-level="6.1.4" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#run-wgcna"><i class="fa fa-check"></i><b>6.1.4</b> <strong>Run WGCNA</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html"><i class="fa fa-check"></i><b>7</b> WGCNA Visualizations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-phenotype-correlation-plot"><i class="fa fa-check"></i><b>7.1</b> <strong>Module-phenotype Correlation Plot</strong></a></li>
<li class="chapter" data-level="7.2" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#samples-dendrogram-plot"><i class="fa fa-check"></i><b>7.2</b> <strong>Samples Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.3" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#cluster-dendrogram-plot"><i class="fa fa-check"></i><b>7.3</b> <strong>Cluster Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.4" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-correlated-features-across-all-modules"><i class="fa fa-check"></i><b>7.4</b> <strong>TOM plot (Heatmap of Correlated Features Across All Modules)</strong></a></li>
<li class="chapter" data-level="7.5" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#mds-plot"><i class="fa fa-check"></i><b>7.5</b> <strong>MDS plot</strong></a></li>
<li class="chapter" data-level="7.6" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-for-selected-features"><i class="fa fa-check"></i><b>7.6</b> <strong>TOM plot for Selected Features</strong></a></li>
<li class="chapter" data-level="7.7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-membership-vs-gene-significance-scatter-plot"><i class="fa fa-check"></i><b>7.7</b> <strong>Module Membership Vs Gene Significance Scatter plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html"><i class="fa fa-check"></i><b>8</b> Module Eigengene Differentiation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#function-for-module-eigengene-differentiation"><i class="fa fa-check"></i><b>8.1</b> <strong>Function for Module eigengene differentiation</strong></a></li>
<li class="chapter" data-level="8.2" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#horizontal-bar-plot"><i class="fa fa-check"></i><b>8.2</b> <strong>Horizontal Bar plot</strong></a></li>
<li class="chapter" data-level="8.3" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#chord-plot-of-hubgenes-and-corresponding-modules"><i class="fa fa-check"></i><b>8.3</b> <strong>Chord plot of hubgenes and corresponding modules</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-enrichment" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Section 5</span> Functional Enrichment<a href="functional-enrichment.html#functional-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p><strong>Description</strong>: This pipeline generates dot plot visualizations for the enriched pathways. The results can be obtained from applying either Metabolite Set Enrichment Analysis (MSEA) or Over-Representation Analysis (ORA) performed on the MetaboAnalyst online platform.</p>
</blockquote>
<p><strong>Project Initialization</strong>:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="functional-enrichment.html#cb67-1" tabindex="-1"></a><span class="co">#Sets the working directory and creates subfolders for organizing outputs.</span></span>
<span id="cb67-2"><a href="functional-enrichment.html#cb67-2" tabindex="-1"></a>mypath<span class="ot">=</span> <span class="st">&quot;C:/Users/USER/Documents/Github/CRC_project/&quot;</span></span>
<span id="cb67-3"><a href="functional-enrichment.html#cb67-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;output&quot;</span>)</span>
<span id="cb67-4"><a href="functional-enrichment.html#cb67-4" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;plots&quot;</span>)</span>
<span id="cb67-5"><a href="functional-enrichment.html#cb67-5" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;input&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="functional-enrichment.html#cb68-1" tabindex="-1"></a><span class="co">#Load libraries</span></span>
<span id="cb68-2"><a href="functional-enrichment.html#cb68-2" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb68-3"><a href="functional-enrichment.html#cb68-3" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb68-4"><a href="functional-enrichment.html#cb68-4" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb68-5"><a href="functional-enrichment.html#cb68-5" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb68-6"><a href="functional-enrichment.html#cb68-6" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb68-7"><a href="functional-enrichment.html#cb68-7" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb68-8"><a href="functional-enrichment.html#cb68-8" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="functional-enrichment.html#cb69-1" tabindex="-1"></a><span class="co"># The enrichment was applied using Metaboanalyst online website</span></span>
<span id="cb69-2"><a href="functional-enrichment.html#cb69-2" tabindex="-1"></a><span class="co"># load enrichment results</span></span>
<span id="cb69-3"><a href="functional-enrichment.html#cb69-3" tabindex="-1"></a>enrich_table <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;input/enrichment_table.csv&quot;</span>))</span>
<span id="cb69-4"><a href="functional-enrichment.html#cb69-4" tabindex="-1"></a><span class="fu">colnames</span>(enrich_table)</span></code></pre></div>
<pre><code>##  [1] &quot;new_pathway_name&quot; &quot;Total.Cmpd&quot;       &quot;Hits&quot;             &quot;Statistic.Q&quot;     
##  [5] &quot;Expected.Q&quot;       &quot;Raw.p&quot;            &quot;Holm.p&quot;           &quot;FDR&quot;             
##  [9] &quot;fold_enrichment&quot;  &quot;P.adj.FDR.log&quot;    &quot;signif&quot;</code></pre>
<div id="data-preparation" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> <strong>Data Preparation</strong><a href="functional-enrichment.html#data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="functional-enrichment.html#cb71-1" tabindex="-1"></a><span class="co"># add necessary columns for plotting</span></span>
<span id="cb71-2"><a href="functional-enrichment.html#cb71-2" tabindex="-1"></a><span class="co">#enrich_table$fold_enrichment &lt;-  enrich_table$hits /enrich_table$total</span></span>
<span id="cb71-3"><a href="functional-enrichment.html#cb71-3" tabindex="-1"></a>enrich_table<span class="sc">$</span>fold_enrichment <span class="ot">&lt;-</span>  enrich_table<span class="sc">$</span>Hits <span class="sc">/</span>enrich_table<span class="sc">$</span>Total.Cmpd</span>
<span id="cb71-4"><a href="functional-enrichment.html#cb71-4" tabindex="-1"></a><span class="co">#enrich_table$fold_enrichment &lt;- round(enrich_table$fold_enrichment,2)</span></span>
<span id="cb71-5"><a href="functional-enrichment.html#cb71-5" tabindex="-1"></a>enrich_table<span class="sc">$</span>FDR.log <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(enrich_table<span class="sc">$</span>FDR)</span>
<span id="cb71-6"><a href="functional-enrichment.html#cb71-6" tabindex="-1"></a>enrich_table<span class="sc">$</span>signif <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(enrich_table<span class="sc">$</span>P.adj.FDR.log <span class="sc">&gt;=</span> <span class="sc">-</span><span class="fu">log10</span>(.<span class="dv">05</span>), <span class="st">&quot;sig&quot;</span>, <span class="st">&quot;non&quot;</span>)</span>
<span id="cb71-7"><a href="functional-enrichment.html#cb71-7" tabindex="-1"></a><span class="fu">table</span>( enrich_table<span class="sc">$</span>signif) </span></code></pre></div>
<pre><code>## 
## sig 
##  39</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="functional-enrichment.html#cb73-1" tabindex="-1"></a><span class="co"># adjust pathway names</span></span>
<span id="cb73-2"><a href="functional-enrichment.html#cb73-2" tabindex="-1"></a><span class="fu">colnames</span>(enrich_table)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;new_pathway_name&quot;</span></span>
<span id="cb73-3"><a href="functional-enrichment.html#cb73-3" tabindex="-1"></a>enrich_table<span class="sc">$</span>new_pathway_name<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;%.*&quot;</span>, <span class="st">&quot;&quot;</span>, enrich_table<span class="sc">$</span>new_pathway_name)</span>
<span id="cb73-4"><a href="functional-enrichment.html#cb73-4" tabindex="-1"></a></span>
<span id="cb73-5"><a href="functional-enrichment.html#cb73-5" tabindex="-1"></a><span class="co"># remove duplicated pathways</span></span>
<span id="cb73-6"><a href="functional-enrichment.html#cb73-6" tabindex="-1"></a>enrich_table <span class="ot">&lt;-</span> enrich_table[<span class="sc">!</span><span class="fu">duplicated</span>(enrich_table<span class="sc">$</span>new_pathway_name),]</span>
<span id="cb73-7"><a href="functional-enrichment.html#cb73-7" tabindex="-1"></a><span class="co"># plot only significant pathways</span></span>
<span id="cb73-8"><a href="functional-enrichment.html#cb73-8" tabindex="-1"></a>enrich_table <span class="ot">&lt;-</span> enrich_table[enrich_table<span class="sc">$</span>signif<span class="sc">==</span> <span class="st">&quot;sig&quot;</span>,]</span></code></pre></div>
</div>
<div id="enrichment-plot" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> <strong>Enrichment plot</strong><a href="functional-enrichment.html#enrichment-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="functional-enrichment.html#cb74-1" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>enrich_table, <span class="fu">aes</span>( <span class="at">y =</span> <span class="fu">reorder</span>(new_pathway_name, FDR.log), </span>
<span id="cb74-2"><a href="functional-enrichment.html#cb74-2" tabindex="-1"></a>                                        <span class="at">x =</span> FDR.log, </span>
<span id="cb74-3"><a href="functional-enrichment.html#cb74-3" tabindex="-1"></a>                                        <span class="at">size =</span> fold_enrichment, </span>
<span id="cb74-4"><a href="functional-enrichment.html#cb74-4" tabindex="-1"></a>                                        <span class="at">color =</span> FDR )) <span class="sc">+</span></span>
<span id="cb74-5"><a href="functional-enrichment.html#cb74-5" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb74-6"><a href="functional-enrichment.html#cb74-6" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">2</span>), <span class="at">breaks =</span> <span class="fu">c</span>( <span class="fl">0.03</span>, <span class="fl">0.07</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span>)</span>
<span id="cb74-7"><a href="functional-enrichment.html#cb74-7" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb74-8"><a href="functional-enrichment.html#cb74-8" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;red&quot;</span>, <span class="at">high =</span> <span class="st">&quot;yellow&quot;</span>, <span class="at">name =</span> <span class="st">&quot;FDR&quot;</span>) <span class="sc">+</span></span>
<span id="cb74-9"><a href="functional-enrichment.html#cb74-9" tabindex="-1"></a>  <span class="co">#breaks = c(min(enrich_table$FDR), 0.25,0.50, 0.75, max(enrich_table$FDR)),</span></span>
<span id="cb74-10"><a href="functional-enrichment.html#cb74-10" tabindex="-1"></a>  <span class="co">#labels = c( round(min(enrich_table$FDR),2), 0.25, &quot;0.50&quot;, 0.75, round(max(enrich_table$FDR),2))) +  # Customize the color scale </span></span>
<span id="cb74-11"><a href="functional-enrichment.html#cb74-11" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Hits/Total&quot;</span>) <span class="sc">+</span></span>
<span id="cb74-12"><a href="functional-enrichment.html#cb74-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;-log10(FDR)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb74-13"><a href="functional-enrichment.html#cb74-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Pathway&quot;</span>) <span class="sc">+</span></span>
<span id="cb74-14"><a href="functional-enrichment.html#cb74-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb74-15"><a href="functional-enrichment.html#cb74-15" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;right&quot;</span>, <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb74-16"><a href="functional-enrichment.html#cb74-16" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb74-17"><a href="functional-enrichment.html#cb74-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="dv">10</span>))</span>
<span id="cb74-18"><a href="functional-enrichment.html#cb74-18" tabindex="-1"></a> </span>
<span id="cb74-19"><a href="functional-enrichment.html#cb74-19" tabindex="-1"></a></span>
<span id="cb74-20"><a href="functional-enrichment.html#cb74-20" tabindex="-1"></a><span class="fu">print</span>(plot1)</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="functional-enrichment.html#cb75-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/enrichment_plot.jpg&quot;</span>), plot1, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
<pre><code>## R version 4.4.1 (2024-06-14 ucrt)
## Platform: x86_64-w64-mingw32/x64
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## 
## locale:
## [1] LC_COLLATE=English_United States.utf8 
## [2] LC_CTYPE=English_United States.utf8   
## [3] LC_MONETARY=English_United States.utf8
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.utf8    
## 
## time zone: Africa/Cairo
## tzcode source: internal
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] ggrepel_0.9.6         viridis_0.6.5         fields_16.2          
##  [4] viridisLite_0.4.2     spam_2.10-0           biomaRt_2.61.2       
##  [7] ComplexHeatmap_2.21.0 circlize_0.4.16       RColorBrewer_1.1-3   
## [10] memoise_2.0.1         caret_6.0-94          lattice_0.22-6       
## [13] pls_2.8-3             Rserve_1.8-13         MetaboAnalystR_3.2.0 
## [16] cowplot_1.1.3         DT_0.33               openxlsx_4.2.6.1     
## [19] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.1        
## [22] purrr_1.0.2           readr_2.1.5           tidyr_1.3.1          
## [25] ggplot2_3.5.1         tidyverse_2.0.0       dplyr_1.1.4          
## [28] plyr_1.8.9            tibble_3.2.1         
## 
## loaded via a namespace (and not attached):
##   [1] splines_4.4.1           filelock_1.0.3          bitops_1.0-7           
##   [4] hardhat_1.4.0           pROC_1.18.5             rpart_4.1.23           
##   [7] httr2_1.0.2             lifecycle_1.0.4         edgeR_4.3.5            
##  [10] doParallel_1.0.17       globals_0.16.3          MASS_7.3-60.2          
##  [13] scrime_1.3.5            crosstalk_1.2.1         magrittr_2.0.3         
##  [16] limma_3.61.5            plotly_4.10.4           sass_0.4.9             
##  [19] rmarkdown_2.27          jquerylib_0.1.4         yaml_2.3.10            
##  [22] zip_2.3.1               DBI_1.2.3               maps_3.4.2             
##  [25] zlibbioc_1.50.0         BiocGenerics_0.52.0     nnet_7.3-19            
##  [28] rappdirs_0.3.3          ipred_0.9-15            GenomeInfoDbData_1.2.12
##  [31] lava_1.8.0              IRanges_2.38.1          S4Vectors_0.42.1       
##  [34] listenv_0.9.1           ellipse_0.5.0           parallelly_1.38.0      
##  [37] codetools_0.2-20        xml2_1.3.6              RApiSerialize_0.1.3    
##  [40] tidyselect_1.2.1        shape_1.4.6.1           UCSC.utils_1.1.0       
##  [43] farver_2.1.2            BiocFileCache_2.13.0    matrixStats_1.3.0      
##  [46] stats4_4.4.1            jsonlite_1.8.8          GetoptLong_1.0.5       
##  [49] multtest_2.61.0         e1071_1.7-14            survival_3.6-4         
##  [52] iterators_1.0.14        systemfonts_1.2.3       foreach_1.5.2          
##  [55] progress_1.2.3          tools_4.4.1             ragg_1.3.2             
##  [58] Rcpp_1.0.13             glue_1.7.0              gridExtra_2.3          
##  [61] prodlim_2024.06.25      xfun_0.46               GenomeInfoDb_1.41.1    
##  [64] crmn_0.0.21             withr_3.0.0             fastmap_1.2.0          
##  [67] caTools_1.18.2          digest_0.6.36           timechange_0.3.0       
##  [70] R6_2.5.1                textshaping_0.4.0       colorspace_2.1-1       
##  [73] Cairo_1.6-2             gtools_3.9.5            RSQLite_2.3.7          
##  [76] generics_0.1.3          data.table_1.15.4       recipes_1.1.0          
##  [79] class_7.3-22            prettyunits_1.2.0       httr_1.4.7             
##  [82] htmlwidgets_1.6.4       ModelMetrics_1.2.2.2    pkgconfig_2.0.3        
##  [85] gtable_0.3.5            timeDate_4032.109       blob_1.2.4             
##  [88] siggenes_1.79.0         impute_1.79.0           XVector_0.44.0         
##  [91] htmltools_0.5.8.1       dotCall64_1.1-1         bookdown_0.40          
##  [94] fgsea_1.31.0            clue_0.3-65             scales_1.3.0           
##  [97] Biobase_2.64.0          png_0.1-8               gower_1.0.1            
## [100] knitr_1.48              rstudioapi_0.16.0       tzdb_0.4.0             
## [103] reshape2_1.4.4          rjson_0.2.21            curl_5.2.1             
## [106] nlme_3.1-164            proxy_0.4-27            cachem_1.1.0           
## [109] GlobalOptions_0.1.2     KernSmooth_2.23-24      parallel_4.4.1         
## [112] AnnotationDbi_1.67.0    pillar_1.11.0           vctrs_0.6.5            
## [115] gplots_3.1.3.1          pcaMethods_1.97.0       stringfish_0.16.0      
## [118] dbplyr_2.5.0            cluster_2.1.6           evaluate_0.24.0        
## [121] magick_2.8.4            cli_3.6.3               locfit_1.5-9.10        
## [124] compiler_4.4.1          rlang_1.1.4             crayon_1.5.3           
## [127] future.apply_1.11.2     labeling_0.4.3          stringi_1.8.4          
## [130] BiocParallel_1.39.0     munsell_0.5.1           Biostrings_2.72.1      
## [133] lazyeval_0.2.2          Matrix_1.7-0            hms_1.1.3              
## [136] glasso_1.11             bit64_4.0.5             future_1.33.2          
## [139] KEGGREST_1.45.1         statmod_1.5.0           highr_0.11             
## [142] qs_0.26.3               igraph_2.0.3            RcppParallel_5.1.8     
## [145] bslib_0.8.0             fastmatch_1.1-4         bit_4.0.5</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="downstream-visualizations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wgcna-pipeline.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/Menna-Arafat/CRC_project/blob/master/4_Enrichment_visualization.rmd",
"text": null
},
"download": ["mybook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
