<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 8 Module Eigengene Differentiation | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</title>
  <meta name="description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 8 Module Eigengene Differentiation | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="github-repo" content="https://github.com/Menna-Arafat/CRC_project" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 8 Module Eigengene Differentiation | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  
  <meta name="twitter:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  

<meta name="author" content="Menna Arafat" />


<meta name="date" content="2025-08-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="wgcna-visualizations.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network-Based Metabolomics Analysis in R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html"><i class="fa fa-check"></i><b>1</b> Metabolomic Raw data Processing</a>
<ul>
<li class="chapter" data-level="1.1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#internal-functions"><i class="fa fa-check"></i><b>1.1</b> <strong>Internal Functions</strong></a></li>
<li class="chapter" data-level="1.2" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#wrapper-function"><i class="fa fa-check"></i><b>1.2</b> <strong>Wrapper Function</strong></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html"><i class="fa fa-check"></i><b>2</b> Dimensionality Reduction Techniques</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#input-data-preparation-for-metaboanalyst"><i class="fa fa-check"></i><b>2.1</b> <strong>Input data Preparation for Metaboanalyst</strong></a></li>
<li class="chapter" data-level="2.2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pca"><i class="fa fa-check"></i><b>2.2</b> <strong>PCA</strong></a></li>
<li class="chapter" data-level="2.3" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pls-da"><i class="fa fa-check"></i><b>2.3</b> <strong>PLS-DA</strong></a></li>
<li class="chapter" data-level="2.4" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#splsda"><i class="fa fa-check"></i><b>2.4</b> <strong>splsda</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html"><i class="fa fa-check"></i><b>3</b> Metaboanalyst Pipeline</a>
<ul>
<li class="chapter" data-level="3.1" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#normality-assessment"><i class="fa fa-check"></i><b>3.1</b> <strong>Normality Assessment</strong></a></li>
<li class="chapter" data-level="3.2" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#metaboanalystr-object-initialization"><i class="fa fa-check"></i><b>3.2</b> <strong>MetaboAnalystR Object Initialization</strong></a></li>
<li class="chapter" data-level="3.3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#auto-scaleing-z-score-normalization"><i class="fa fa-check"></i><b>3.3</b> <strong>Auto-scaleing (z-score Normalization)</strong></a></li>
<li class="chapter" data-level="3.4" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#post-normalization-normality-check"><i class="fa fa-check"></i><b>3.4</b> <strong>Post-Normalization Normality Check</strong></a></li>
<li class="chapter" data-level="3.5" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.5</b> <strong>Fold change Analysis</strong></a></li>
<li class="chapter" data-level="3.6" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.6</b> <strong>Differential Expression Analysis</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html"><i class="fa fa-check"></i><b>4</b> Downstream Visualizations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#volcano-plot"><i class="fa fa-check"></i><b>4.1</b> <strong>Volcano Plot</strong></a></li>
<li class="chapter" data-level="4.2" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#heatmap-plot"><i class="fa fa-check"></i><b>4.2</b> <strong>Heatmap Plot</strong></a></li>
<li class="chapter" data-level="4.3" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#box-plots"><i class="fa fa-check"></i><b>4.3</b> <strong>Box Plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment.html"><a href="functional-enrichment.html"><i class="fa fa-check"></i><b>5</b> Functional Enrichment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment.html"><a href="functional-enrichment.html#data-preparation"><i class="fa fa-check"></i><b>5.1</b> <strong>Data Preparation</strong></a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment.html"><a href="functional-enrichment.html#enrichment-plot"><i class="fa fa-check"></i><b>5.2</b> <strong>Enrichment plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html"><i class="fa fa-check"></i><b>6</b> WGCNA Pipeline</a>
<ul>
<li class="chapter" data-level="6.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-pipeline-1"><i class="fa fa-check"></i><b>6.1</b> <strong>WGCNA Pipeline</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#choose-appropriate-power-for-soft-thresholding"><i class="fa fa-check"></i><b>6.1.1</b> <strong>Choose appropriate power for soft-thresholding</strong></a></li>
<li class="chapter" data-level="6.1.2" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#set-main-parameters"><i class="fa fa-check"></i><b>6.1.2</b> <strong>Set Main Parameters</strong></a></li>
<li class="chapter" data-level="6.1.3" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-main-function"><i class="fa fa-check"></i><b>6.1.3</b> <strong>WGCNA Main Function</strong></a></li>
<li class="chapter" data-level="6.1.4" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#run-wgcna"><i class="fa fa-check"></i><b>6.1.4</b> <strong>Run WGCNA</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html"><i class="fa fa-check"></i><b>7</b> WGCNA Visualizations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-phenotype-correlation-plot"><i class="fa fa-check"></i><b>7.1</b> <strong>Module-phenotype Correlation Plot</strong></a></li>
<li class="chapter" data-level="7.2" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#samples-dendrogram-plot"><i class="fa fa-check"></i><b>7.2</b> <strong>Samples Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.3" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#cluster-dendrogram-plot"><i class="fa fa-check"></i><b>7.3</b> <strong>Cluster Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.4" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-correlated-proteins-across-all-modules"><i class="fa fa-check"></i><b>7.4</b> <strong>TOM plot (heatmap of correlated proteins across all modules)</strong></a></li>
<li class="chapter" data-level="7.5" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#mds-plot"><i class="fa fa-check"></i><b>7.5</b> <strong>MDS plot</strong></a></li>
<li class="chapter" data-level="7.6" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-modules-for-selected-proteins-based-on-fold-change-or-de"><i class="fa fa-check"></i><b>7.6</b> <strong>TOM plot/ heatmap of modules for selected proteins (based on fold change or DE)</strong></a></li>
<li class="chapter" data-level="7.7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-membership-vs-gene-significance-scatter-plot"><i class="fa fa-check"></i><b>7.7</b> <strong>Module Membership Vs Gene Significance Scatter plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html"><i class="fa fa-check"></i><b>8</b> Module Eigengene Differentiation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#function-for-module-eigengene-differentiation"><i class="fa fa-check"></i><b>8.1</b> <strong>Function for Module eigengene differentiation</strong></a></li>
<li class="chapter" data-level="8.2" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#horizontal-bar-plot"><i class="fa fa-check"></i><b>8.2</b> <strong>Horizontal Bar plot</strong></a></li>
<li class="chapter" data-level="8.3" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#chord-plot-of-hubgenes-and-corresponding-modules"><i class="fa fa-check"></i><b>8.3</b> <strong>Chord plot of hubgenes and corresponding modules</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-eigengene-differentiation" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Section 8</span> Module Eigengene Differentiation<a href="module-eigengene-differentiation.html#module-eigengene-differentiation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p><strong>Description</strong>: This pipeline performs post-WGCNA analysis by testing whether module eigengenes significantly differentiate between phenotypic groups.It uses adaptive statistical testing (parametric or non-parametric) based on normality and number of groups to assess eigengene–phenotype associations. And The results are corrected for multiple testing using FDR Benjamini-Hochberg test.</p>
</blockquote>
<p><strong>Project Initialization</strong></p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="module-eigengene-differentiation.html#cb134-1" tabindex="-1"></a><span class="co">#Sets the working directory and creates subfolders for organizing outputs.</span></span>
<span id="cb134-2"><a href="module-eigengene-differentiation.html#cb134-2" tabindex="-1"></a>mypath<span class="ot">=</span> <span class="st">&quot;C:/Users/USER/Documents/Github/CRC_project/&quot;</span></span>
<span id="cb134-3"><a href="module-eigengene-differentiation.html#cb134-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;output&quot;</span>)</span>
<span id="cb134-4"><a href="module-eigengene-differentiation.html#cb134-4" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;plots&quot;</span>)</span>
<span id="cb134-5"><a href="module-eigengene-differentiation.html#cb134-5" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;input&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="module-eigengene-differentiation.html#cb135-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb135-2"><a href="module-eigengene-differentiation.html#cb135-2" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb135-3"><a href="module-eigengene-differentiation.html#cb135-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb135-4"><a href="module-eigengene-differentiation.html#cb135-4" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb135-5"><a href="module-eigengene-differentiation.html#cb135-5" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb135-6"><a href="module-eigengene-differentiation.html#cb135-6" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb135-7"><a href="module-eigengene-differentiation.html#cb135-7" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb135-8"><a href="module-eigengene-differentiation.html#cb135-8" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb135-9"><a href="module-eigengene-differentiation.html#cb135-9" tabindex="-1"></a><span class="fu">library</span>(gplots)</span>
<span id="cb135-10"><a href="module-eigengene-differentiation.html#cb135-10" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="module-eigengene-differentiation.html#cb136-1" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb136-2"><a href="module-eigengene-differentiation.html#cb136-2" tabindex="-1"></a>module_eigengenes<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/module_eigengenes.csv&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;X&quot;</span>) </span>
<span id="cb136-3"><a href="module-eigengene-differentiation.html#cb136-3" tabindex="-1"></a>data<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;input/data_for_downstream.csv&quot;</span>)) </span>
<span id="cb136-4"><a href="module-eigengene-differentiation.html#cb136-4" tabindex="-1"></a>data <span class="ot">=</span> data <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="fu">colnames</span>(data)[<span class="dv">1</span>])</span>
<span id="cb136-5"><a href="module-eigengene-differentiation.html#cb136-5" tabindex="-1"></a>dir<span class="ot">=</span> <span class="fu">paste0</span>(mypath, <span class="st">&quot;output/&quot;</span>)</span>
<span id="cb136-6"><a href="module-eigengene-differentiation.html#cb136-6" tabindex="-1"></a>group_dist<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(data))</span>
<span id="cb136-7"><a href="module-eigengene-differentiation.html#cb136-7" tabindex="-1"></a>group_levels<span class="ot">=</span> <span class="fu">unique</span>(group_dist)</span></code></pre></div>
<div id="function-for-module-eigengene-differentiation" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> <strong>Function for Module eigengene differentiation</strong><a href="module-eigengene-differentiation.html#function-for-module-eigengene-differentiation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="module-eigengene-differentiation.html#cb137-1" tabindex="-1"></a><span class="co"># A function that apply the appropriate statistical test based on </span></span>
<span id="cb137-2"><a href="module-eigengene-differentiation.html#cb137-2" tabindex="-1"></a><span class="co"># normality and number of groups to assess differential expression </span></span>
<span id="cb137-3"><a href="module-eigengene-differentiation.html#cb137-3" tabindex="-1"></a><span class="co"># of module eigengenes (first principal components) between groups.</span></span>
<span id="cb137-4"><a href="module-eigengene-differentiation.html#cb137-4" tabindex="-1"></a>test_differentiation <span class="ot">&lt;-</span> <span class="cf">function</span>(module_eigengenes, group_vector, <span class="at">output_path =</span> <span class="cn">NULL</span>) {</span>
<span id="cb137-5"><a href="module-eigengene-differentiation.html#cb137-5" tabindex="-1"></a> </span>
<span id="cb137-6"><a href="module-eigengene-differentiation.html#cb137-6" tabindex="-1"></a>  check_normality <span class="ot">&lt;-</span> <span class="cf">function</span>(vector) {</span>
<span id="cb137-7"><a href="module-eigengene-differentiation.html#cb137-7" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(vector)</span>
<span id="cb137-8"><a href="module-eigengene-differentiation.html#cb137-8" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">is.na</span>(x)]</span>
<span id="cb137-9"><a href="module-eigengene-differentiation.html#cb137-9" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(x) <span class="sc">&lt;=</span> <span class="dv">3</span>) {</span>
<span id="cb137-10"><a href="module-eigengene-differentiation.html#cb137-10" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">&quot;Non-parametric&quot;</span>)  </span>
<span id="cb137-11"><a href="module-eigengene-differentiation.html#cb137-11" tabindex="-1"></a>    }</span>
<span id="cb137-12"><a href="module-eigengene-differentiation.html#cb137-12" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">shapiro.test</span>(x), <span class="at">error =</span> <span class="cf">function</span>(e) <span class="fu">return</span>(<span class="cn">NULL</span>))</span>
<span id="cb137-13"><a href="module-eigengene-differentiation.html#cb137-13" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(result)) <span class="fu">return</span>(<span class="st">&quot;Non-parametric&quot;</span>)</span>
<span id="cb137-14"><a href="module-eigengene-differentiation.html#cb137-14" tabindex="-1"></a>    <span class="cf">if</span> (result<span class="sc">$</span>p.value <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="st">&quot;Parametric&quot;</span> <span class="cf">else</span> <span class="st">&quot;Non-parametric&quot;</span></span>
<span id="cb137-15"><a href="module-eigengene-differentiation.html#cb137-15" tabindex="-1"></a>  }</span>
<span id="cb137-16"><a href="module-eigengene-differentiation.html#cb137-16" tabindex="-1"></a>  </span>
<span id="cb137-17"><a href="module-eigengene-differentiation.html#cb137-17" tabindex="-1"></a>  <span class="co"># Prepare group information</span></span>
<span id="cb137-18"><a href="module-eigengene-differentiation.html#cb137-18" tabindex="-1"></a>  group_vector <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(group_vector)</span>
<span id="cb137-19"><a href="module-eigengene-differentiation.html#cb137-19" tabindex="-1"></a>  group_levels <span class="ot">&lt;-</span> <span class="fu">levels</span>(group_vector)</span>
<span id="cb137-20"><a href="module-eigengene-differentiation.html#cb137-20" tabindex="-1"></a>  n_groups <span class="ot">&lt;-</span> <span class="fu">length</span>(group_levels)</span>
<span id="cb137-21"><a href="module-eigengene-differentiation.html#cb137-21" tabindex="-1"></a>  mods <span class="ot">&lt;-</span> <span class="fu">colnames</span>(module_eigengenes)</span>
<span id="cb137-22"><a href="module-eigengene-differentiation.html#cb137-22" tabindex="-1"></a>  </span>
<span id="cb137-23"><a href="module-eigengene-differentiation.html#cb137-23" tabindex="-1"></a>  results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb137-24"><a href="module-eigengene-differentiation.html#cb137-24" tabindex="-1"></a>  </span>
<span id="cb137-25"><a href="module-eigengene-differentiation.html#cb137-25" tabindex="-1"></a>  <span class="cf">for</span> (mod <span class="cf">in</span> mods) {</span>
<span id="cb137-26"><a href="module-eigengene-differentiation.html#cb137-26" tabindex="-1"></a>    x <span class="ot">&lt;-</span> module_eigengenes[[mod]]</span>
<span id="cb137-27"><a href="module-eigengene-differentiation.html#cb137-27" tabindex="-1"></a>    </span>
<span id="cb137-28"><a href="module-eigengene-differentiation.html#cb137-28" tabindex="-1"></a>    test_type <span class="ot">&lt;-</span> <span class="fu">check_normality</span>(x)</span>
<span id="cb137-29"><a href="module-eigengene-differentiation.html#cb137-29" tabindex="-1"></a>    </span>
<span id="cb137-30"><a href="module-eigengene-differentiation.html#cb137-30" tabindex="-1"></a>    <span class="cf">if</span> (n_groups <span class="sc">==</span> <span class="dv">2</span>) {</span>
<span id="cb137-31"><a href="module-eigengene-differentiation.html#cb137-31" tabindex="-1"></a>      <span class="cf">if</span> (test_type <span class="sc">==</span> <span class="st">&quot;Parametric&quot;</span>) {</span>
<span id="cb137-32"><a href="module-eigengene-differentiation.html#cb137-32" tabindex="-1"></a>        res <span class="ot">&lt;-</span> <span class="fu">t.test</span>(x <span class="sc">~</span> group_vector)</span>
<span id="cb137-33"><a href="module-eigengene-differentiation.html#cb137-33" tabindex="-1"></a>        test_used <span class="ot">&lt;-</span> <span class="st">&quot;t-test&quot;</span></span>
<span id="cb137-34"><a href="module-eigengene-differentiation.html#cb137-34" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb137-35"><a href="module-eigengene-differentiation.html#cb137-35" tabindex="-1"></a>        res <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(x <span class="sc">~</span> group_vector)</span>
<span id="cb137-36"><a href="module-eigengene-differentiation.html#cb137-36" tabindex="-1"></a>        test_used <span class="ot">&lt;-</span> <span class="st">&quot;Wilcoxon&quot;</span></span>
<span id="cb137-37"><a href="module-eigengene-differentiation.html#cb137-37" tabindex="-1"></a>      }</span>
<span id="cb137-38"><a href="module-eigengene-differentiation.html#cb137-38" tabindex="-1"></a>      p <span class="ot">&lt;-</span> res<span class="sc">$</span>p.value</span>
<span id="cb137-39"><a href="module-eigengene-differentiation.html#cb137-39" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (n_groups <span class="sc">&gt;</span> <span class="dv">2</span>) {</span>
<span id="cb137-40"><a href="module-eigengene-differentiation.html#cb137-40" tabindex="-1"></a>      <span class="cf">if</span> (test_type <span class="sc">==</span> <span class="st">&quot;Parametric&quot;</span>) {</span>
<span id="cb137-41"><a href="module-eigengene-differentiation.html#cb137-41" tabindex="-1"></a>        res <span class="ot">&lt;-</span> <span class="fu">aov</span>(x <span class="sc">~</span> group_vector)</span>
<span id="cb137-42"><a href="module-eigengene-differentiation.html#cb137-42" tabindex="-1"></a>        p <span class="ot">&lt;-</span> <span class="fu">summary</span>(res)[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb137-43"><a href="module-eigengene-differentiation.html#cb137-43" tabindex="-1"></a>        test_used <span class="ot">&lt;-</span> <span class="st">&quot;ANOVA&quot;</span></span>
<span id="cb137-44"><a href="module-eigengene-differentiation.html#cb137-44" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb137-45"><a href="module-eigengene-differentiation.html#cb137-45" tabindex="-1"></a>        res <span class="ot">&lt;-</span> <span class="fu">kruskal.test</span>(x <span class="sc">~</span> group_vector)</span>
<span id="cb137-46"><a href="module-eigengene-differentiation.html#cb137-46" tabindex="-1"></a>        p <span class="ot">&lt;-</span> res<span class="sc">$</span>p.value</span>
<span id="cb137-47"><a href="module-eigengene-differentiation.html#cb137-47" tabindex="-1"></a>        test_used <span class="ot">&lt;-</span> <span class="st">&quot;Kruskal-Wallis&quot;</span></span>
<span id="cb137-48"><a href="module-eigengene-differentiation.html#cb137-48" tabindex="-1"></a>      }</span>
<span id="cb137-49"><a href="module-eigengene-differentiation.html#cb137-49" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb137-50"><a href="module-eigengene-differentiation.html#cb137-50" tabindex="-1"></a>      p <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb137-51"><a href="module-eigengene-differentiation.html#cb137-51" tabindex="-1"></a>      test_used <span class="ot">&lt;-</span> <span class="st">&quot;Invalid group size&quot;</span></span>
<span id="cb137-52"><a href="module-eigengene-differentiation.html#cb137-52" tabindex="-1"></a>    }</span>
<span id="cb137-53"><a href="module-eigengene-differentiation.html#cb137-53" tabindex="-1"></a>    </span>
<span id="cb137-54"><a href="module-eigengene-differentiation.html#cb137-54" tabindex="-1"></a>    results_list[[mod]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">p.value =</span> p, <span class="at">test =</span> test_used)</span>
<span id="cb137-55"><a href="module-eigengene-differentiation.html#cb137-55" tabindex="-1"></a>  }</span>
<span id="cb137-56"><a href="module-eigengene-differentiation.html#cb137-56" tabindex="-1"></a>  </span>
<span id="cb137-57"><a href="module-eigengene-differentiation.html#cb137-57" tabindex="-1"></a>  <span class="co"># Convert results to a data frame</span></span>
<span id="cb137-58"><a href="module-eigengene-differentiation.html#cb137-58" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="fu">names</span>(results_list), <span class="cf">function</span>(mod) {</span>
<span id="cb137-59"><a href="module-eigengene-differentiation.html#cb137-59" tabindex="-1"></a>    row <span class="ot">&lt;-</span> results_list[[mod]]</span>
<span id="cb137-60"><a href="module-eigengene-differentiation.html#cb137-60" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">modules =</span> mod, <span class="at">p.value =</span> row<span class="sc">$</span>p.value, <span class="at">test =</span> row<span class="sc">$</span>test)</span>
<span id="cb137-61"><a href="module-eigengene-differentiation.html#cb137-61" tabindex="-1"></a>  }))</span>
<span id="cb137-62"><a href="module-eigengene-differentiation.html#cb137-62" tabindex="-1"></a>  </span>
<span id="cb137-63"><a href="module-eigengene-differentiation.html#cb137-63" tabindex="-1"></a>  <span class="co"># Adjust FDR</span></span>
<span id="cb137-64"><a href="module-eigengene-differentiation.html#cb137-64" tabindex="-1"></a>  res_df<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res_df<span class="sc">$</span>p.value, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb137-65"><a href="module-eigengene-differentiation.html#cb137-65" tabindex="-1"></a>  res_df<span class="sc">$</span>sig <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(res_df<span class="sc">$</span>fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">&quot;***&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb137-66"><a href="module-eigengene-differentiation.html#cb137-66" tabindex="-1"></a>  res_df<span class="sc">$</span>module <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, res_df<span class="sc">$</span>module)</span>
<span id="cb137-67"><a href="module-eigengene-differentiation.html#cb137-67" tabindex="-1"></a>  </span>
<span id="cb137-68"><a href="module-eigengene-differentiation.html#cb137-68" tabindex="-1"></a>  <span class="co"># Save if path is provided</span></span>
<span id="cb137-69"><a href="module-eigengene-differentiation.html#cb137-69" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(output_path)) {</span>
<span id="cb137-70"><a href="module-eigengene-differentiation.html#cb137-70" tabindex="-1"></a>    <span class="fu">write.csv</span>(res_df,<span class="fu">paste0</span>(output_path ,<span class="st">&quot;ME_differentiation.csv&quot;</span>) , <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb137-71"><a href="module-eigengene-differentiation.html#cb137-71" tabindex="-1"></a>  }</span>
<span id="cb137-72"><a href="module-eigengene-differentiation.html#cb137-72" tabindex="-1"></a>  </span>
<span id="cb137-73"><a href="module-eigengene-differentiation.html#cb137-73" tabindex="-1"></a>  <span class="fu">return</span>(res_df)</span>
<span id="cb137-74"><a href="module-eigengene-differentiation.html#cb137-74" tabindex="-1"></a>}</span>
<span id="cb137-75"><a href="module-eigengene-differentiation.html#cb137-75" tabindex="-1"></a></span>
<span id="cb137-76"><a href="module-eigengene-differentiation.html#cb137-76" tabindex="-1"></a>res<span class="ot">=</span> <span class="fu">test_differentiation</span>(module_eigengenes, group_dist, dir)</span>
<span id="cb137-77"><a href="module-eigengene-differentiation.html#cb137-77" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<pre><code>##       modules      p.value     test          fdr sig    module
## 1      MEblue 1.082509e-05 Wilcoxon 2.165018e-05 ***      blue
## 2     MEbrown 7.959363e-01 Wilcoxon 7.959363e-01         brown
## 3 MEturquoise 1.082509e-05 Wilcoxon 2.165018e-05 *** turquoise
## 4      MEgrey 3.930481e-01 Wilcoxon 5.240642e-01          grey</code></pre>
</div>
<div id="horizontal-bar-plot" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> <strong>Horizontal Bar plot</strong><a href="module-eigengene-differentiation.html#horizontal-bar-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="module-eigengene-differentiation.html#cb139-1" tabindex="-1"></a>p<span class="ot">=</span> <span class="fu">ggplot</span>(res) <span class="sc">+</span></span>
<span id="cb139-2"><a href="module-eigengene-differentiation.html#cb139-2" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> modules, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">fill =</span> module), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb139-3"><a href="module-eigengene-differentiation.html#cb139-3" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="co"># Flip coordinates to make the bar plot horizontal</span></span>
<span id="cb139-4"><a href="module-eigengene-differentiation.html#cb139-4" tabindex="-1"></a>  <span class="fu">scale_fill_identity</span>() <span class="sc">+</span> <span class="co"># Use actual colors specified in the data frame</span></span>
<span id="cb139-5"><a href="module-eigengene-differentiation.html#cb139-5" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span>  modules  , <span class="at">y =</span><span class="sc">-</span><span class="fu">log10</span>(fdr) , <span class="at">label =</span> sig), </span>
<span id="cb139-6"><a href="module-eigengene-differentiation.html#cb139-6" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size=</span>.<span class="dv">5</span> ,<span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb139-7"><a href="module-eigengene-differentiation.html#cb139-7" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb139-8"><a href="module-eigengene-differentiation.html#cb139-8" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;-log10(FDR)&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Modules&quot;</span>) <span class="sc">+</span></span>
<span id="cb139-9"><a href="module-eigengene-differentiation.html#cb139-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Module Eigengene Differentiation&quot;</span>) <span class="sc">+</span> <span class="co"># Label axes and provide a title</span></span>
<span id="cb139-10"><a href="module-eigengene-differentiation.html#cb139-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb139-11"><a href="module-eigengene-differentiation.html#cb139-11" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb139-12"><a href="module-eigengene-differentiation.html#cb139-12" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb139-13"><a href="module-eigengene-differentiation.html#cb139-13" tabindex="-1"></a>        <span class="at">axis.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">13</span>) ) <span class="sc">+</span> </span>
<span id="cb139-14"><a href="module-eigengene-differentiation.html#cb139-14" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb139-15"><a href="module-eigengene-differentiation.html#cb139-15" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">lwd=</span> <span class="dv">1</span>)</span>
<span id="cb139-16"><a href="module-eigengene-differentiation.html#cb139-16" tabindex="-1"></a></span>
<span id="cb139-17"><a href="module-eigengene-differentiation.html#cb139-17" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-71-1.png" width="672" /></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="module-eigengene-differentiation.html#cb140-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/module_differentiation.png&quot;</span>), p,<span class="at">width=</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">7</span>, <span class="at">dpi=</span><span class="dv">900</span> )</span></code></pre></div>
</div>
<div id="chord-plot-of-hubgenes-and-corresponding-modules" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> <strong>Chord plot of hubgenes and corresponding modules</strong><a href="module-eigengene-differentiation.html#chord-plot-of-hubgenes-and-corresponding-modules" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="module-eigengene-differentiation.html#cb141-1" tabindex="-1"></a>module_hubs<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/module_hubs.csv&quot;</span>)) </span>
<span id="cb141-2"><a href="module-eigengene-differentiation.html#cb141-2" tabindex="-1"></a><span class="fu">colnames</span>(module_hubs)<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">&quot;_hub&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">names</span>(module_hubs) )</span>
<span id="cb141-3"><a href="module-eigengene-differentiation.html#cb141-3" tabindex="-1"></a><span class="co"># We can plot only the hub features of significant modules (Module differentiation output)</span></span>
<span id="cb141-4"><a href="module-eigengene-differentiation.html#cb141-4" tabindex="-1"></a>module_sig<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/ME_differentiation.csv&quot;</span>)) </span>
<span id="cb141-5"><a href="module-eigengene-differentiation.html#cb141-5" tabindex="-1"></a>m.sig<span class="ot">=</span> module_sig<span class="sc">$</span>module[module_sig<span class="sc">$</span>sig<span class="sc">==</span> <span class="st">&quot;***&quot;</span>]</span>
<span id="cb141-6"><a href="module-eigengene-differentiation.html#cb141-6" tabindex="-1"></a>module_hubs<span class="ot">=</span> module_hubs <span class="sc">|&gt;</span>  dplyr<span class="sc">::</span><span class="fu">select</span>(m.sig)</span>
<span id="cb141-7"><a href="module-eigengene-differentiation.html#cb141-7" tabindex="-1"></a></span>
<span id="cb141-8"><a href="module-eigengene-differentiation.html#cb141-8" tabindex="-1"></a><span class="co">#build similarity/ design matrix</span></span>
<span id="cb141-9"><a href="module-eigengene-differentiation.html#cb141-9" tabindex="-1"></a>keydrivers<span class="ot">=</span> <span class="fu">unlist</span>(module_hubs) <span class="sc">|&gt;</span>  <span class="fu">unique</span>() </span>
<span id="cb141-10"><a href="module-eigengene-differentiation.html#cb141-10" tabindex="-1"></a>keydrivers<span class="ot">=</span> keydrivers[keydrivers <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(keydrivers)]</span>
<span id="cb141-11"><a href="module-eigengene-differentiation.html#cb141-11" tabindex="-1"></a>mtx<span class="ot">=</span> <span class="fu">matrix</span>(<span class="at">nrow=</span> <span class="fu">ncol</span>(module_hubs), <span class="at">ncol =</span> <span class="fu">length</span>( keydrivers)) </span>
<span id="cb141-12"><a href="module-eigengene-differentiation.html#cb141-12" tabindex="-1"></a><span class="fu">row.names</span>(mtx)<span class="ot">=</span> <span class="fu">colnames</span>(module_hubs)</span>
<span id="cb141-13"><a href="module-eigengene-differentiation.html#cb141-13" tabindex="-1"></a><span class="fu">colnames</span>(mtx)<span class="ot">=</span> <span class="fu">paste0</span>(keydrivers)</span>
<span id="cb141-14"><a href="module-eigengene-differentiation.html#cb141-14" tabindex="-1"></a></span>
<span id="cb141-15"><a href="module-eigengene-differentiation.html#cb141-15" tabindex="-1"></a><span class="co">#build similarity matrix </span></span>
<span id="cb141-16"><a href="module-eigengene-differentiation.html#cb141-16" tabindex="-1"></a><span class="co">#colnames of matrix included in keydrivers specified for certain module/ phenotype(rows) then put in 1</span></span>
<span id="cb141-17"><a href="module-eigengene-differentiation.html#cb141-17" tabindex="-1"></a>mod<span class="ot">=</span> <span class="fu">apply</span>(module_hubs, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.list</span>(x))</span>
<span id="cb141-18"><a href="module-eigengene-differentiation.html#cb141-18" tabindex="-1"></a></span>
<span id="cb141-19"><a href="module-eigengene-differentiation.html#cb141-19" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(mod)){</span>
<span id="cb141-20"><a href="module-eigengene-differentiation.html#cb141-20" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(mtx)) {</span>
<span id="cb141-21"><a href="module-eigengene-differentiation.html#cb141-21" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">colnames</span>(mtx)[j] <span class="sc">%in%</span> mod[[i]] ) {</span>
<span id="cb141-22"><a href="module-eigengene-differentiation.html#cb141-22" tabindex="-1"></a>      mtx[i, j] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb141-23"><a href="module-eigengene-differentiation.html#cb141-23" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb141-24"><a href="module-eigengene-differentiation.html#cb141-24" tabindex="-1"></a>      mtx[i, j] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb141-25"><a href="module-eigengene-differentiation.html#cb141-25" tabindex="-1"></a>    }</span>
<span id="cb141-26"><a href="module-eigengene-differentiation.html#cb141-26" tabindex="-1"></a>  }</span>
<span id="cb141-27"><a href="module-eigengene-differentiation.html#cb141-27" tabindex="-1"></a>}</span>
<span id="cb141-28"><a href="module-eigengene-differentiation.html#cb141-28" tabindex="-1"></a></span>
<span id="cb141-29"><a href="module-eigengene-differentiation.html#cb141-29" tabindex="-1"></a><span class="co"># make shorter row names</span></span>
<span id="cb141-30"><a href="module-eigengene-differentiation.html#cb141-30" tabindex="-1"></a><span class="fu">colnames</span>(mtx)<span class="ot">=</span>  <span class="fu">substr</span>(<span class="fu">colnames</span>(mtx), <span class="dv">1</span>, <span class="dv">25</span>) </span>
<span id="cb141-31"><a href="module-eigengene-differentiation.html#cb141-31" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb141-32"><a href="module-eigengene-differentiation.html#cb141-32" tabindex="-1"></a></span>
<span id="cb141-33"><a href="module-eigengene-differentiation.html#cb141-33" tabindex="-1"></a><span class="co"># Define chord plot function</span></span>
<span id="cb141-34"><a href="module-eigengene-differentiation.html#cb141-34" tabindex="-1"></a>plot_chord_hubs <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb141-35"><a href="module-eigengene-differentiation.html#cb141-35" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb141-36"><a href="module-eigengene-differentiation.html#cb141-36" tabindex="-1"></a>  <span class="fu">circos.par</span>(</span>
<span id="cb141-37"><a href="module-eigengene-differentiation.html#cb141-37" tabindex="-1"></a>    <span class="at">gap.degree =</span> <span class="dv">1</span>,</span>
<span id="cb141-38"><a href="module-eigengene-differentiation.html#cb141-38" tabindex="-1"></a>    <span class="at">track.margin =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>),</span>
<span id="cb141-39"><a href="module-eigengene-differentiation.html#cb141-39" tabindex="-1"></a>    <span class="at">canvas.xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.2</span>),</span>
<span id="cb141-40"><a href="module-eigengene-differentiation.html#cb141-40" tabindex="-1"></a>    <span class="at">canvas.ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.2</span>),</span>
<span id="cb141-41"><a href="module-eigengene-differentiation.html#cb141-41" tabindex="-1"></a>    <span class="at">points.overflow.warning =</span> <span class="cn">FALSE</span></span>
<span id="cb141-42"><a href="module-eigengene-differentiation.html#cb141-42" tabindex="-1"></a>  )</span>
<span id="cb141-43"><a href="module-eigengene-differentiation.html#cb141-43" tabindex="-1"></a>  </span>
<span id="cb141-44"><a href="module-eigengene-differentiation.html#cb141-44" tabindex="-1"></a>  <span class="fu">chordDiagram</span>(</span>
<span id="cb141-45"><a href="module-eigengene-differentiation.html#cb141-45" tabindex="-1"></a>    mtx,</span>
<span id="cb141-46"><a href="module-eigengene-differentiation.html#cb141-46" tabindex="-1"></a>    <span class="at">annotationTrack =</span> <span class="st">&quot;grid&quot;</span>,</span>
<span id="cb141-47"><a href="module-eigengene-differentiation.html#cb141-47" tabindex="-1"></a>    <span class="at">transparency =</span> <span class="fl">0.5</span></span>
<span id="cb141-48"><a href="module-eigengene-differentiation.html#cb141-48" tabindex="-1"></a>  )</span>
<span id="cb141-49"><a href="module-eigengene-differentiation.html#cb141-49" tabindex="-1"></a>  </span>
<span id="cb141-50"><a href="module-eigengene-differentiation.html#cb141-50" tabindex="-1"></a>  <span class="co"># Labels customization</span></span>
<span id="cb141-51"><a href="module-eigengene-differentiation.html#cb141-51" tabindex="-1"></a>  labels_to_asterisk <span class="ot">&lt;-</span> <span class="cn">NULL</span>  <span class="co"># Example: c(&quot;TP53&quot;, &quot;MYC&quot;)</span></span>
<span id="cb141-52"><a href="module-eigengene-differentiation.html#cb141-52" tabindex="-1"></a>  labels_red <span class="ot">&lt;-</span> <span class="cn">NULL</span>          <span class="co"># Example: c(&quot;BRCA1&quot;, &quot;EGFR&quot;)</span></span>
<span id="cb141-53"><a href="module-eigengene-differentiation.html#cb141-53" tabindex="-1"></a>  </span>
<span id="cb141-54"><a href="module-eigengene-differentiation.html#cb141-54" tabindex="-1"></a>  <span class="fu">circos.track</span>(<span class="at">track.index =</span> <span class="dv">1</span>, <span class="at">panel.fun =</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb141-55"><a href="module-eigengene-differentiation.html#cb141-55" tabindex="-1"></a>    label <span class="ot">&lt;-</span> CELL_META<span class="sc">$</span>sector.index</span>
<span id="cb141-56"><a href="module-eigengene-differentiation.html#cb141-56" tabindex="-1"></a>    modified_label <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(label <span class="sc">%in%</span> labels_to_asterisk,</span>
<span id="cb141-57"><a href="module-eigengene-differentiation.html#cb141-57" tabindex="-1"></a>                             <span class="fu">paste0</span>(label, <span class="st">&quot; ***&quot;</span>),</span>
<span id="cb141-58"><a href="module-eigengene-differentiation.html#cb141-58" tabindex="-1"></a>                             label)</span>
<span id="cb141-59"><a href="module-eigengene-differentiation.html#cb141-59" tabindex="-1"></a>    label_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(label <span class="sc">%in%</span> labels_red, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;black&quot;</span>)</span>
<span id="cb141-60"><a href="module-eigengene-differentiation.html#cb141-60" tabindex="-1"></a>    </span>
<span id="cb141-61"><a href="module-eigengene-differentiation.html#cb141-61" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb141-62"><a href="module-eigengene-differentiation.html#cb141-62" tabindex="-1"></a>      CELL_META<span class="sc">$</span>xcenter,</span>
<span id="cb141-63"><a href="module-eigengene-differentiation.html#cb141-63" tabindex="-1"></a>      CELL_META<span class="sc">$</span>cell.ylim[<span class="dv">2</span>] <span class="sc">*</span> <span class="fl">3.5</span>,</span>
<span id="cb141-64"><a href="module-eigengene-differentiation.html#cb141-64" tabindex="-1"></a>      modified_label,</span>
<span id="cb141-65"><a href="module-eigengene-differentiation.html#cb141-65" tabindex="-1"></a>      <span class="at">col =</span> label_color,</span>
<span id="cb141-66"><a href="module-eigengene-differentiation.html#cb141-66" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">0.7</span>,</span>
<span id="cb141-67"><a href="module-eigengene-differentiation.html#cb141-67" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb141-68"><a href="module-eigengene-differentiation.html#cb141-68" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">&quot;clockwise&quot;</span>,</span>
<span id="cb141-69"><a href="module-eigengene-differentiation.html#cb141-69" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb141-70"><a href="module-eigengene-differentiation.html#cb141-70" tabindex="-1"></a>      <span class="at">adj =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb141-71"><a href="module-eigengene-differentiation.html#cb141-71" tabindex="-1"></a>    )</span>
<span id="cb141-72"><a href="module-eigengene-differentiation.html#cb141-72" tabindex="-1"></a>  }, <span class="at">bg.border =</span> <span class="cn">NA</span>)</span>
<span id="cb141-73"><a href="module-eigengene-differentiation.html#cb141-73" tabindex="-1"></a>  </span>
<span id="cb141-74"><a href="module-eigengene-differentiation.html#cb141-74" tabindex="-1"></a>  <span class="fu">circos.clear</span>()</span>
<span id="cb141-75"><a href="module-eigengene-differentiation.html#cb141-75" tabindex="-1"></a>}</span>
<span id="cb141-76"><a href="module-eigengene-differentiation.html#cb141-76" tabindex="-1"></a></span>
<span id="cb141-77"><a href="module-eigengene-differentiation.html#cb141-77" tabindex="-1"></a><span class="co"># Display</span></span>
<span id="cb141-78"><a href="module-eigengene-differentiation.html#cb141-78" tabindex="-1"></a><span class="fu">plot_chord_hubs</span>()</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="module-eigengene-differentiation.html#cb142-1" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb142-2"><a href="module-eigengene-differentiation.html#cb142-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/chord_plot_hubs.png&quot;</span>), <span class="at">width =</span> <span class="dv">9000</span>, <span class="at">height =</span> <span class="dv">9000</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb142-3"><a href="module-eigengene-differentiation.html#cb142-3" tabindex="-1"></a><span class="fu">plot_chord_hubs</span>()</span>
<span id="cb142-4"><a href="module-eigengene-differentiation.html#cb142-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
<pre><code>## R version 4.4.1 (2024-06-14 ucrt)
## Platform: x86_64-w64-mingw32/x64
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## 
## locale:
## [1] LC_COLLATE=English_United States.utf8 
## [2] LC_CTYPE=English_United States.utf8   
## [3] LC_MONETARY=English_United States.utf8
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.utf8    
## 
## time zone: Africa/Cairo
## tzcode source: internal
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] gplots_3.1.3.1        gridExtra_2.3         flashClust_1.01-2    
##  [4] ggdendro_0.2.0        ape_5.8               pROC_1.18.5          
##  [7] gtools_3.9.5          WGCNA_1.72-5          fastcluster_1.2.6    
## [10] dynamicTreeCut_1.63-1 ggrepel_0.9.6         viridis_0.6.5        
## [13] fields_16.2           viridisLite_0.4.2     spam_2.10-0          
## [16] biomaRt_2.61.2        ComplexHeatmap_2.21.0 circlize_0.4.16      
## [19] RColorBrewer_1.1-3    memoise_2.0.1         caret_6.0-94         
## [22] lattice_0.22-6        pls_2.8-3             Rserve_1.8-13        
## [25] MetaboAnalystR_3.2.0  cowplot_1.1.3         DT_0.33              
## [28] openxlsx_4.2.6.1      lubridate_1.9.3       forcats_1.0.0        
## [31] stringr_1.5.1         purrr_1.0.2           readr_2.1.5          
## [34] tidyr_1.3.1           ggplot2_3.5.1         tidyverse_2.0.0      
## [37] dplyr_1.1.4           plyr_1.8.9            tibble_3.2.1         
## 
## loaded via a namespace (and not attached):
##   [1] matrixStats_1.3.0       bitops_1.0-7            httr_1.4.7             
##   [4] doParallel_1.0.17       tools_4.4.1             backports_1.5.0        
##   [7] R6_2.5.1                lazyeval_0.2.2          GetoptLong_1.0.5       
##  [10] withr_3.0.0             prettyunits_1.2.0       preprocessCore_1.67.0  
##  [13] cli_3.6.3               Biobase_2.64.0          textshaping_0.4.0      
##  [16] Cairo_1.6-2             labeling_0.4.3          sass_0.4.9             
##  [19] proxy_0.4-27            systemfonts_1.2.3       foreign_0.8-86         
##  [22] siggenes_1.79.0         parallelly_1.38.0       scrime_1.3.5           
##  [25] maps_3.4.2              limma_3.61.5            rstudioapi_0.16.0      
##  [28] impute_1.79.0           RSQLite_2.3.7           generics_0.1.3         
##  [31] shape_1.4.6.1           RApiSerialize_0.1.3     crmn_0.0.21            
##  [34] crosstalk_1.2.1         zip_2.3.1               GO.db_3.19.1           
##  [37] Matrix_1.7-0            S4Vectors_0.42.1        lifecycle_1.0.4        
##  [40] yaml_2.3.10             edgeR_4.3.5             recipes_1.1.0          
##  [43] BiocFileCache_2.13.0    blob_1.2.4              crayon_1.5.3           
##  [46] KEGGREST_1.45.1         magick_2.8.4            pillar_1.11.0          
##  [49] knitr_1.48              fgsea_1.31.0            rjson_0.2.21           
##  [52] future.apply_1.11.2     codetools_0.2-20        fastmatch_1.1-4        
##  [55] glue_1.7.0              pcaMethods_1.97.0       data.table_1.15.4      
##  [58] vctrs_0.6.5             png_0.1-8               gtable_0.3.5           
##  [61] cachem_1.1.0            gower_1.0.1             xfun_0.46              
##  [64] prodlim_2024.06.25      survival_3.6-4          timeDate_4032.109      
##  [67] iterators_1.0.14        hardhat_1.4.0           lava_1.8.0             
##  [70] statmod_1.5.0           ipred_0.9-15            nlme_3.1-164           
##  [73] bit64_4.0.5             progress_1.2.3          filelock_1.0.3         
##  [76] GenomeInfoDb_1.41.1     bslib_0.8.0             KernSmooth_2.23-24     
##  [79] rpart_4.1.23            colorspace_2.1-1        BiocGenerics_0.52.0    
##  [82] DBI_1.2.3               Hmisc_5.1-3             nnet_7.3-19            
##  [85] tidyselect_1.2.1        bit_4.0.5               compiler_4.4.1         
##  [88] curl_5.2.1              httr2_1.0.2             htmlTable_2.4.3        
##  [91] xml2_1.3.6              plotly_4.10.4           stringfish_0.16.0      
##  [94] bookdown_0.40           checkmate_2.3.1         scales_1.3.0           
##  [97] caTools_1.18.2          rappdirs_0.3.3          digest_0.6.36          
## [100] rmarkdown_2.27          XVector_0.44.0          htmltools_0.5.8.1      
## [103] pkgconfig_2.0.3         base64enc_0.1-3         highr_0.11             
## [106] dbplyr_2.5.0            fastmap_1.2.0           rlang_1.1.4            
## [109] GlobalOptions_0.1.2     htmlwidgets_1.6.4       UCSC.utils_1.1.0       
## [112] farver_2.1.2            jquerylib_0.1.4         jsonlite_1.8.8         
## [115] BiocParallel_1.39.0     ModelMetrics_1.2.2.2    magrittr_2.0.3         
## [118] Formula_1.2-5           GenomeInfoDbData_1.2.12 dotCall64_1.1-1        
## [121] munsell_0.5.1           Rcpp_1.0.13             stringi_1.8.4          
## [124] zlibbioc_1.50.0         MASS_7.3-60.2           parallel_4.4.1         
## [127] listenv_0.9.1           Biostrings_2.72.1       splines_4.4.1          
## [130] multtest_2.61.0         hms_1.1.3               locfit_1.5-9.10        
## [133] igraph_2.0.3            reshape2_1.4.4          stats4_4.4.1           
## [136] evaluate_0.24.0         RcppParallel_5.1.8      tzdb_0.4.0             
## [139] foreach_1.5.2           qs_0.26.3               future_1.33.2          
## [142] clue_0.3-65             e1071_1.7-14            glasso_1.11            
## [145] class_7.3-22            ragg_1.3.2              AnnotationDbi_1.67.0   
## [148] ellipse_0.5.0           IRanges_2.38.1          cluster_2.1.6          
## [151] timechange_0.3.0        globals_0.16.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="wgcna-visualizations.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/Menna-Arafat/CRC_project/7_Module_differentiation.rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["mybook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
