<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 6 WGCNA Pipeline | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</title>
  <meta name="description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 6 WGCNA Pipeline | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="github-repo" content="https://github.com/Menna-Arafat/CRC_project" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 6 WGCNA Pipeline | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  
  <meta name="twitter:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  

<meta name="author" content="Menna Arafat" />


<meta name="date" content="2025-08-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="functional-enrichment.html"/>
<link rel="next" href="wgcna-visualizations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network-Based Metabolomics Analysis in R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html"><i class="fa fa-check"></i><b>1</b> Metabolomic Raw data Processing</a>
<ul>
<li class="chapter" data-level="1.1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#internal-functions"><i class="fa fa-check"></i><b>1.1</b> <strong>Internal Functions</strong></a></li>
<li class="chapter" data-level="1.2" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#wrapper-function"><i class="fa fa-check"></i><b>1.2</b> <strong>Wrapper Function</strong></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html"><i class="fa fa-check"></i><b>2</b> Dimensionality Reduction Techniques</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#input-data-preparation-for-metaboanalyst"><i class="fa fa-check"></i><b>2.1</b> <strong>Input data Preparation for Metaboanalyst</strong></a></li>
<li class="chapter" data-level="2.2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pca"><i class="fa fa-check"></i><b>2.2</b> <strong>PCA</strong></a></li>
<li class="chapter" data-level="2.3" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pls-da"><i class="fa fa-check"></i><b>2.3</b> <strong>PLS-DA</strong></a></li>
<li class="chapter" data-level="2.4" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#splsda"><i class="fa fa-check"></i><b>2.4</b> <strong>splsda</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html"><i class="fa fa-check"></i><b>3</b> Metaboanalyst Pipeline</a>
<ul>
<li class="chapter" data-level="3.1" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#normality-assessment"><i class="fa fa-check"></i><b>3.1</b> <strong>Normality Assessment</strong></a></li>
<li class="chapter" data-level="3.2" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#metaboanalystr-object-initialization"><i class="fa fa-check"></i><b>3.2</b> <strong>MetaboAnalystR Object Initialization</strong></a></li>
<li class="chapter" data-level="3.3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#auto-scaleing-z-score-normalization"><i class="fa fa-check"></i><b>3.3</b> <strong>Auto-scaleing (z-score Normalization)</strong></a></li>
<li class="chapter" data-level="3.4" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#post-normalization-normality-check"><i class="fa fa-check"></i><b>3.4</b> <strong>Post-Normalization Normality Check</strong></a></li>
<li class="chapter" data-level="3.5" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.5</b> <strong>Fold change Analysis</strong></a></li>
<li class="chapter" data-level="3.6" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.6</b> <strong>Differential Expression Analysis</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html"><i class="fa fa-check"></i><b>4</b> Downstream Visualizations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#volcano-plot"><i class="fa fa-check"></i><b>4.1</b> <strong>Volcano Plot</strong></a></li>
<li class="chapter" data-level="4.2" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#heatmap-plot"><i class="fa fa-check"></i><b>4.2</b> <strong>Heatmap Plot</strong></a></li>
<li class="chapter" data-level="4.3" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#box-plots"><i class="fa fa-check"></i><b>4.3</b> <strong>Box Plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment.html"><a href="functional-enrichment.html"><i class="fa fa-check"></i><b>5</b> Functional Enrichment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment.html"><a href="functional-enrichment.html#data-preparation"><i class="fa fa-check"></i><b>5.1</b> <strong>Data Preparation</strong></a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment.html"><a href="functional-enrichment.html#enrichment-plot"><i class="fa fa-check"></i><b>5.2</b> <strong>Enrichment plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html"><i class="fa fa-check"></i><b>6</b> WGCNA Pipeline</a>
<ul>
<li class="chapter" data-level="6.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-pipeline-1"><i class="fa fa-check"></i><b>6.1</b> <strong>WGCNA Pipeline</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#choose-appropriate-power-for-soft-thresholding"><i class="fa fa-check"></i><b>6.1.1</b> <strong>Choose appropriate power for soft-thresholding</strong></a></li>
<li class="chapter" data-level="6.1.2" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#set-main-parameters"><i class="fa fa-check"></i><b>6.1.2</b> <strong>Set Main Parameters</strong></a></li>
<li class="chapter" data-level="6.1.3" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-main-function"><i class="fa fa-check"></i><b>6.1.3</b> <strong>WGCNA Main Function</strong></a></li>
<li class="chapter" data-level="6.1.4" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#run-wgcna"><i class="fa fa-check"></i><b>6.1.4</b> <strong>Run WGCNA</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html"><i class="fa fa-check"></i><b>7</b> WGCNA Visualizations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-phenotype-correlation-plot"><i class="fa fa-check"></i><b>7.1</b> <strong>Module-phenotype Correlation Plot</strong></a></li>
<li class="chapter" data-level="7.2" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#samples-dendrogram-plot"><i class="fa fa-check"></i><b>7.2</b> <strong>Samples Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.3" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#cluster-dendrogram-plot"><i class="fa fa-check"></i><b>7.3</b> <strong>Cluster Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.4" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-correlated-proteins-across-all-modules"><i class="fa fa-check"></i><b>7.4</b> <strong>TOM plot (heatmap of correlated proteins across all modules)</strong></a></li>
<li class="chapter" data-level="7.5" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#mds-plot"><i class="fa fa-check"></i><b>7.5</b> <strong>MDS plot</strong></a></li>
<li class="chapter" data-level="7.6" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-modules-for-selected-proteins-based-on-fold-change-or-de"><i class="fa fa-check"></i><b>7.6</b> <strong>TOM plot/ heatmap of modules for selected proteins (based on fold change or DE)</strong></a></li>
<li class="chapter" data-level="7.7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-membership-vs-gene-significance-scatter-plot"><i class="fa fa-check"></i><b>7.7</b> <strong>Module Membership Vs Gene Significance Scatter plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html"><i class="fa fa-check"></i><b>8</b> Module Eigengene Differentiation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#function-for-module-eigengene-differentiation"><i class="fa fa-check"></i><b>8.1</b> <strong>Function for Module eigengene differentiation</strong></a></li>
<li class="chapter" data-level="8.2" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#horizontal-bar-plot"><i class="fa fa-check"></i><b>8.2</b> <strong>Horizontal Bar plot</strong></a></li>
<li class="chapter" data-level="8.3" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#chord-plot-of-hubgenes-and-corresponding-modules"><i class="fa fa-check"></i><b>8.3</b> <strong>Chord plot of hubgenes and corresponding modules</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wgcna-pipeline" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Section 6</span> WGCNA Pipeline<a href="wgcna-pipeline.html#wgcna-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p><strong>Description</strong>: This pipeline automate WGCNA pipeline.
So What is WGCNA? Weighted Gene/Metabolite Co-expression Network Analysis (WGCNA/WMCNA)
WGCNA (for genes) and WMCNA (for metabolites) are network-based methods used to identify groups of co-expressed molecules, called modules, that may share common biological functions.<br />
The analysis starts by computing pairwise correlations (typically Pearson) between all features across samples. These correlations are then transformed into a weighted network using a soft-thresholding power, this help remove weak association- which ensures that the resulting network follows a scale-free topology, a common feature of biological systems.<br />
From the adjacency matrix, a topological overlap matrix (TOM) is calculated to measure how connected each pair of features is, considering both direct and shared connections. This leads to the identification of modules through hierarchical clustering and dynamic tree cutting, grouping together molecules with similar expression patterns.<br />
To relate modules to biological phenotypes, the module eigengenes (the first principal component of each module) are correlated with external traits, typically using Pearson correlation. When the phenotype is binary, it is encoded as 0 and 1. Modules that are significantly correlated with the phenotype of interest are then subjected to functional enrichment analysis to uncover associated pathways and biological functions.<br />
Finally, within these relevant modules, hub genes or metabolites—those with high intra-modular connectivity (i.e., high degree centrality)—are identified. These hubs are considered potential biomarker candidates or key regulatory features associated with the condition under study.<br />
Output Files:<br />
<strong>module_eigengenes.csv</strong>: Eigengenes (first principal component) for each module; used for trait correlation.<br />
<strong>gene_module_assignment.csv</strong>: Module assignment of each gene/metabolite (e.g., blue, brown, etc.).<br />
<strong>modules.csv</strong>: Lists of features grouped by module. Each column represents a module.<br />
<strong>module_membership.csv</strong>: Correlation (kME) of each gene with its module eigengene (i.e., intramodular connectivity).<br />
<strong>gene.trait.corr.csv</strong>: Correlation between each gene and the phenotype(s).<br />
<strong>module_hubs.csv</strong>: Hub genes/metabolites within each module (high kME &gt; hub_threshold).<br />
<strong>CytoscapeInput-edges-.txt</strong>: Edge list for each module’s network (TOM similarity &gt; 0.02); ready for Cytoscape.</p>
</blockquote>
<p><strong>Project Initialization</strong></p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="wgcna-pipeline.html#cb77-1" tabindex="-1"></a><span class="co">#Sets the working directory and creates subfolders for organizing outputs.</span></span>
<span id="cb77-2"><a href="wgcna-pipeline.html#cb77-2" tabindex="-1"></a>mypath<span class="ot">=</span> <span class="st">&quot;C:/Users/USER/Documents/Github/CRC_project/&quot;</span></span>
<span id="cb77-3"><a href="wgcna-pipeline.html#cb77-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;output&quot;</span>)</span>
<span id="cb77-4"><a href="wgcna-pipeline.html#cb77-4" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;plots&quot;</span>)</span>
<span id="cb77-5"><a href="wgcna-pipeline.html#cb77-5" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;input&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="wgcna-pipeline.html#cb78-1" tabindex="-1"></a><span class="co">#Load libraries</span></span>
<span id="cb78-2"><a href="wgcna-pipeline.html#cb78-2" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb78-3"><a href="wgcna-pipeline.html#cb78-3" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb78-4"><a href="wgcna-pipeline.html#cb78-4" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb78-5"><a href="wgcna-pipeline.html#cb78-5" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb78-6"><a href="wgcna-pipeline.html#cb78-6" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb78-7"><a href="wgcna-pipeline.html#cb78-7" tabindex="-1"></a><span class="fu">allowWGCNAThreads</span>()   </span></code></pre></div>
<pre><code>## Allowing multi-threading with up to 4 threads.</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="wgcna-pipeline.html#cb80-1" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb80-2"><a href="wgcna-pipeline.html#cb80-2" tabindex="-1"></a><span class="co">#expression data</span></span>
<span id="cb80-3"><a href="wgcna-pipeline.html#cb80-3" tabindex="-1"></a>data<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;input/data_for_downstream.csv&quot;</span>))</span>
<span id="cb80-4"><a href="wgcna-pipeline.html#cb80-4" tabindex="-1"></a>data <span class="ot">=</span> data <span class="sc">|&gt;</span>  <span class="fu">column_to_rownames</span>(<span class="fu">colnames</span>(data)[<span class="dv">1</span>]) </span>
<span id="cb80-5"><a href="wgcna-pipeline.html#cb80-5" tabindex="-1"></a></span>
<span id="cb80-6"><a href="wgcna-pipeline.html#cb80-6" tabindex="-1"></a>group_dist<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(data))</span>
<span id="cb80-7"><a href="wgcna-pipeline.html#cb80-7" tabindex="-1"></a>group_levels<span class="ot">=</span> <span class="fu">unique</span>(group_dist)</span>
<span id="cb80-8"><a href="wgcna-pipeline.html#cb80-8" tabindex="-1"></a><span class="co">#metadata</span></span>
<span id="cb80-9"><a href="wgcna-pipeline.html#cb80-9" tabindex="-1"></a>metadata<span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample=</span> <span class="fu">colnames</span>(data) ,</span>
<span id="cb80-10"><a href="wgcna-pipeline.html#cb80-10" tabindex="-1"></a>                     <span class="at">condition=</span> group_dist)</span>
<span id="cb80-11"><a href="wgcna-pipeline.html#cb80-11" tabindex="-1"></a><span class="co">#metadata= read.csv(&quot;input/metadata_test.csv&quot; )</span></span>
<span id="cb80-12"><a href="wgcna-pipeline.html#cb80-12" tabindex="-1"></a>metadata <span class="ot">=</span> metadata <span class="sc">|&gt;</span>   <span class="fu">column_to_rownames</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb80-13"><a href="wgcna-pipeline.html#cb80-13" tabindex="-1"></a>metadata<span class="sc">$</span>condition<span class="ot">=</span> <span class="fu">factor</span>(metadata<span class="sc">$</span>condition, <span class="at">levels =</span> group_levels)</span>
<span id="cb80-14"><a href="wgcna-pipeline.html#cb80-14" tabindex="-1"></a>design<span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span>condition , metadata)</span>
<span id="cb80-15"><a href="wgcna-pipeline.html#cb80-15" tabindex="-1"></a><span class="fu">colnames</span>(design)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(design))</span>
<span id="cb80-16"><a href="wgcna-pipeline.html#cb80-16" tabindex="-1"></a><span class="fu">head</span>(design)</span></code></pre></div>
<pre><code>##        CRC Ctrl
## CRC_01   1    0
## CRC_02   1    0
## CRC_03   1    0
## CRC_04   1    0
## CRC_05   1    0
## CRC_06   1    0</code></pre>
<div id="wgcna-pipeline-1" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> <strong>WGCNA Pipeline</strong><a href="wgcna-pipeline.html#wgcna-pipeline-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="choose-appropriate-power-for-soft-thresholding" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> <strong>Choose appropriate power for soft-thresholding</strong><a href="wgcna-pipeline.html#choose-appropriate-power-for-soft-thresholding" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="wgcna-pipeline.html#cb82-1" tabindex="-1"></a><span class="co"># Elbow plot to choose the appropriate soft-thresholding power that ensures a scale-free (power-law) network topology.</span></span>
<span id="cb82-2"><a href="wgcna-pipeline.html#cb82-2" tabindex="-1"></a>plotelbow <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb82-3"><a href="wgcna-pipeline.html#cb82-3" tabindex="-1"></a></span>
<span id="cb82-4"><a href="wgcna-pipeline.html#cb82-4" tabindex="-1"></a>  <span class="co"># Choose a set of soft-thresholding powers (up to 20 is more informative)</span></span>
<span id="cb82-5"><a href="wgcna-pipeline.html#cb82-5" tabindex="-1"></a>  powers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb82-6"><a href="wgcna-pipeline.html#cb82-6" tabindex="-1"></a>  <span class="co"># Pick the soft threshold based on scale-free topology criterion</span></span>
<span id="cb82-7"><a href="wgcna-pipeline.html#cb82-7" tabindex="-1"></a>  sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(<span class="fu">t</span>(data), <span class="at">powerVector =</span> powers, <span class="at">verbose =</span> <span class="dv">5</span>)</span>
<span id="cb82-8"><a href="wgcna-pipeline.html#cb82-8" tabindex="-1"></a>  <span class="co"># Two plots side by side</span></span>
<span id="cb82-9"><a href="wgcna-pipeline.html#cb82-9" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)) </span>
<span id="cb82-10"><a href="wgcna-pipeline.html#cb82-10" tabindex="-1"></a>  <span class="co"># Plot the Scale-Free Topology Fit Index (R^2)</span></span>
<span id="cb82-11"><a href="wgcna-pipeline.html#cb82-11" tabindex="-1"></a>  <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>], </span>
<span id="cb82-12"><a href="wgcna-pipeline.html#cb82-12" tabindex="-1"></a>       <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb82-13"><a href="wgcna-pipeline.html#cb82-13" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>, </span>
<span id="cb82-14"><a href="wgcna-pipeline.html#cb82-14" tabindex="-1"></a>       <span class="at">ylab =</span> <span class="st">&quot;Scale Free Topology Model Fit, signed R^2&quot;</span>,</span>
<span id="cb82-15"><a href="wgcna-pipeline.html#cb82-15" tabindex="-1"></a>       <span class="at">type =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb82-16"><a href="wgcna-pipeline.html#cb82-16" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">&quot;Scale Independence&quot;</span>)</span>
<span id="cb82-17"><a href="wgcna-pipeline.html#cb82-17" tabindex="-1"></a>  <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>], </span>
<span id="cb82-18"><a href="wgcna-pipeline.html#cb82-18" tabindex="-1"></a>       <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb82-19"><a href="wgcna-pipeline.html#cb82-19" tabindex="-1"></a>       <span class="at">labels =</span> powers, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb82-20"><a href="wgcna-pipeline.html#cb82-20" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.80</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lty =</span> <span class="dv">2</span>)  <span class="co"># R^2 threshold line</span></span>
<span id="cb82-21"><a href="wgcna-pipeline.html#cb82-21" tabindex="-1"></a>  </span>
<span id="cb82-22"><a href="wgcna-pipeline.html#cb82-22" tabindex="-1"></a>  <span class="co"># Plot the Mean Connectivity</span></span>
<span id="cb82-23"><a href="wgcna-pipeline.html#cb82-23" tabindex="-1"></a>  <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>], </span>
<span id="cb82-24"><a href="wgcna-pipeline.html#cb82-24" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>, </span>
<span id="cb82-25"><a href="wgcna-pipeline.html#cb82-25" tabindex="-1"></a>       <span class="at">ylab =</span> <span class="st">&quot;Mean Connectivity&quot;</span>, </span>
<span id="cb82-26"><a href="wgcna-pipeline.html#cb82-26" tabindex="-1"></a>       <span class="at">type =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb82-27"><a href="wgcna-pipeline.html#cb82-27" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">&quot;Mean Connectivity&quot;</span>)</span>
<span id="cb82-28"><a href="wgcna-pipeline.html#cb82-28" tabindex="-1"></a>  <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>], </span>
<span id="cb82-29"><a href="wgcna-pipeline.html#cb82-29" tabindex="-1"></a>       <span class="at">labels =</span> powers, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb82-30"><a href="wgcna-pipeline.html#cb82-30" tabindex="-1"></a>  </span>
<span id="cb82-31"><a href="wgcna-pipeline.html#cb82-31" tabindex="-1"></a>  </span>
<span id="cb82-32"><a href="wgcna-pipeline.html#cb82-32" tabindex="-1"></a>}</span>
<span id="cb82-33"><a href="wgcna-pipeline.html#cb82-33" tabindex="-1"></a></span>
<span id="cb82-34"><a href="wgcna-pipeline.html#cb82-34" tabindex="-1"></a><span class="co">#The chosen power should correspond to Signed R² (scale-free topology fit index) ≥ 0.9 &amp;</span></span>
<span id="cb82-35"><a href="wgcna-pipeline.html#cb82-35" tabindex="-1"></a><span class="co">#Mean connectivity is not too low (network should remain connected)</span></span>
<span id="cb82-36"><a href="wgcna-pipeline.html#cb82-36" tabindex="-1"></a><span class="co"># Display the plot</span></span>
<span id="cb82-37"><a href="wgcna-pipeline.html#cb82-37" tabindex="-1"></a><span class="fu">plotelbow</span>(data)</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 263.
##  pickSoftThreshold: calculating connectivity for given powers...
##    ..working on genes 1 through 263 of 263
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0133 -0.203       -0.25200    83.8     76.00  133.0
## 2      2   0.1260 -3.130       -0.12400    43.9     31.70   94.1
## 3      3   0.1240 -2.570       -0.10400    30.1     16.60   78.5
## 4      4   0.1920 -3.680       -0.03720    24.1     11.00   71.2
## 5      5   0.2060 -3.590       -0.01890    21.0      7.74   67.5
## 6      6   0.2390 -4.180        0.02530    19.2      6.00   65.3
## 7      7   0.1910 -3.630       -0.02780    18.1      4.61   64.0
## 8      8   0.2550 -4.350        0.05880    17.3      3.68   63.2
## 9      9   0.1990 -3.710       -0.01200    16.8      3.06   62.6
## 10    10   0.2090 -3.740       -0.00272    16.4      2.40   62.2</code></pre>
<p><img src="mybook_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="wgcna-pipeline.html#cb84-1" tabindex="-1"></a><span class="co"># save</span></span>
<span id="cb84-2"><a href="wgcna-pipeline.html#cb84-2" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;plots/elbow_plot.png&quot;</span>, <span class="at">width =</span> <span class="dv">1700</span>, <span class="at">height =</span> <span class="dv">1000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb84-3"><a href="wgcna-pipeline.html#cb84-3" tabindex="-1"></a><span class="fu">plotelbow</span>(data)</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 263.
##  pickSoftThreshold: calculating connectivity for given powers...
##    ..working on genes 1 through 263 of 263
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0133 -0.203       -0.25200    83.8     76.00  133.0
## 2      2   0.1260 -3.130       -0.12400    43.9     31.70   94.1
## 3      3   0.1240 -2.570       -0.10400    30.1     16.60   78.5
## 4      4   0.1920 -3.680       -0.03720    24.1     11.00   71.2
## 5      5   0.2060 -3.590       -0.01890    21.0      7.74   67.5
## 6      6   0.2390 -4.180        0.02530    19.2      6.00   65.3
## 7      7   0.1910 -3.630       -0.02780    18.1      4.61   64.0
## 8      8   0.2550 -4.350        0.05880    17.3      3.68   63.2
## 9      9   0.1990 -3.710       -0.01200    16.8      3.06   62.6
## 10    10   0.2090 -3.740       -0.00272    16.4      2.40   62.2</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="wgcna-pipeline.html#cb86-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="set-main-parameters" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> <strong>Set Main Parameters</strong><a href="wgcna-pipeline.html#set-main-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="wgcna-pipeline.html#cb88-1" tabindex="-1"></a><span class="co">#set your parameters</span></span>
<span id="cb88-2"><a href="wgcna-pipeline.html#cb88-2" tabindex="-1"></a>power<span class="ot">=</span> <span class="dv">4</span></span>
<span id="cb88-3"><a href="wgcna-pipeline.html#cb88-3" tabindex="-1"></a>minModuleSize <span class="ot">=</span> <span class="dv">20</span></span>
<span id="cb88-4"><a href="wgcna-pipeline.html#cb88-4" tabindex="-1"></a>metadata_binary<span class="ot">=</span> design</span>
<span id="cb88-5"><a href="wgcna-pipeline.html#cb88-5" tabindex="-1"></a>hub_threshold<span class="ot">=</span> <span class="fl">0.85</span></span>
<span id="cb88-6"><a href="wgcna-pipeline.html#cb88-6" tabindex="-1"></a>GS_threshold<span class="ot">=</span> <span class="fl">0.85</span> <span class="co">#(.2) default</span></span>
<span id="cb88-7"><a href="wgcna-pipeline.html#cb88-7" tabindex="-1"></a><span class="co">#mapping_file= read.table(&quot;resources/Metabolites-HMDB.csv&quot;, header = T)</span></span></code></pre></div>
</div>
<div id="wgcna-main-function" class="section level3 hasAnchor" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> <strong>WGCNA Main Function</strong><a href="wgcna-pipeline.html#wgcna-main-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="wgcna-pipeline.html#cb89-1" tabindex="-1"></a>Run_WGCNA<span class="ot">=</span> <span class="cf">function</span>(data, metadata_binary, power,minModuleSize, hub_threshold ,GS_threshold,</span>
<span id="cb89-2"><a href="wgcna-pipeline.html#cb89-2" tabindex="-1"></a>                     <span class="at">mapping_file=</span><span class="cn">NULL</span>){</span>
<span id="cb89-3"><a href="wgcna-pipeline.html#cb89-3" tabindex="-1"></a>  </span>
<span id="cb89-4"><a href="wgcna-pipeline.html#cb89-4" tabindex="-1"></a>  power<span class="ot">=</span> power</span>
<span id="cb89-5"><a href="wgcna-pipeline.html#cb89-5" tabindex="-1"></a>  minModuleSize<span class="ot">=</span> minModuleSize</span>
<span id="cb89-6"><a href="wgcna-pipeline.html#cb89-6" tabindex="-1"></a>  data[]<span class="ot">=</span> <span class="fu">lapply</span>(data, as.numeric)</span>
<span id="cb89-7"><a href="wgcna-pipeline.html#cb89-7" tabindex="-1"></a>  datExpr<span class="ot">=</span> <span class="fu">t</span>(data) <span class="co">#so that samples become in row</span></span>
<span id="cb89-8"><a href="wgcna-pipeline.html#cb89-8" tabindex="-1"></a>  <span class="co">#This function checks data for missing entries, entries with weights below a threshold, and zero-variance genes, </span></span>
<span id="cb89-9"><a href="wgcna-pipeline.html#cb89-9" tabindex="-1"></a>  goods <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr, <span class="at">verbose =</span> <span class="dv">3</span>)</span>
<span id="cb89-10"><a href="wgcna-pipeline.html#cb89-10" tabindex="-1"></a>  datExpr<span class="ot">=</span> datExpr[goods<span class="sc">$</span>goodSamples<span class="sc">==</span> <span class="cn">TRUE</span>, goods<span class="sc">$</span>goodGenes <span class="sc">==</span> <span class="cn">TRUE</span> ]</span>
<span id="cb89-11"><a href="wgcna-pipeline.html#cb89-11" tabindex="-1"></a>  </span>
<span id="cb89-12"><a href="wgcna-pipeline.html#cb89-12" tabindex="-1"></a>  net <span class="ot">=</span> <span class="fu">blockwiseModules</span>(datExpr, <span class="at">corType =</span> <span class="st">&quot;pearson&quot;</span>, <span class="at">maxBlockSize =</span> <span class="dv">5000</span>, </span>
<span id="cb89-13"><a href="wgcna-pipeline.html#cb89-13" tabindex="-1"></a>                         <span class="at">networkType =</span> <span class="st">&quot;signed&quot;</span>, <span class="at">power =</span> power, <span class="at">minModuleSize =</span>minModuleSize,</span>
<span id="cb89-14"><a href="wgcna-pipeline.html#cb89-14" tabindex="-1"></a>                         <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>, </span>
<span id="cb89-15"><a href="wgcna-pipeline.html#cb89-15" tabindex="-1"></a>                         <span class="at">numericLabels =</span> F, <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb89-16"><a href="wgcna-pipeline.html#cb89-16" tabindex="-1"></a>                         <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>, <span class="at">saveTOMFileBase =</span> <span class="st">&quot;TOM&quot;</span>)</span>
<span id="cb89-17"><a href="wgcna-pipeline.html#cb89-17" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;module detection is done&quot;</span>)</span>
<span id="cb89-18"><a href="wgcna-pipeline.html#cb89-18" tabindex="-1"></a>  </span>
<span id="cb89-19"><a href="wgcna-pipeline.html#cb89-19" tabindex="-1"></a>  <span class="co"># A data frame with module eigengenes </span></span>
<span id="cb89-20"><a href="wgcna-pipeline.html#cb89-20" tabindex="-1"></a>  module_eigengenes <span class="ot">&lt;-</span> net<span class="sc">$</span>MEs</span>
<span id="cb89-21"><a href="wgcna-pipeline.html#cb89-21" tabindex="-1"></a>  <span class="fu">write.csv</span>(module_eigengenes, <span class="st">&quot;output/module_eigengenes.csv&quot;</span>)</span>
<span id="cb89-22"><a href="wgcna-pipeline.html#cb89-22" tabindex="-1"></a>  <span class="fu">names</span>(module_eigengenes)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">names</span>(module_eigengenes) )</span>
<span id="cb89-23"><a href="wgcna-pipeline.html#cb89-23" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;module_eigengenes file is saved&quot;</span>)</span>
<span id="cb89-24"><a href="wgcna-pipeline.html#cb89-24" tabindex="-1"></a>  </span>
<span id="cb89-25"><a href="wgcna-pipeline.html#cb89-25" tabindex="-1"></a>  <span class="co">#gene module assignment</span></span>
<span id="cb89-26"><a href="wgcna-pipeline.html#cb89-26" tabindex="-1"></a>  module.gene.assign<span class="ot">=</span> net<span class="sc">$</span>colors <span class="co">#vector of colors that assign each gene to its corresponding module</span></span>
<span id="cb89-27"><a href="wgcna-pipeline.html#cb89-27" tabindex="-1"></a>  <span class="fu">write.csv</span>(module.gene.assign, <span class="st">&quot;output/gene_module_assignment.csv&quot;</span>)</span>
<span id="cb89-28"><a href="wgcna-pipeline.html#cb89-28" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;gene_module_assignment file is saved&quot;</span>)</span>
<span id="cb89-29"><a href="wgcna-pipeline.html#cb89-29" tabindex="-1"></a>  </span>
<span id="cb89-30"><a href="wgcna-pipeline.html#cb89-30" tabindex="-1"></a>  <span class="co">#Extract proteins for each modules</span></span>
<span id="cb89-31"><a href="wgcna-pipeline.html#cb89-31" tabindex="-1"></a>  module_names<span class="ot">=</span> <span class="fu">names</span>(module_eigengenes)</span>
<span id="cb89-32"><a href="wgcna-pipeline.html#cb89-32" tabindex="-1"></a>  module_lists<span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb89-33"><a href="wgcna-pipeline.html#cb89-33" tabindex="-1"></a>  </span>
<span id="cb89-34"><a href="wgcna-pipeline.html#cb89-34" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> module_names){</span>
<span id="cb89-35"><a href="wgcna-pipeline.html#cb89-35" tabindex="-1"></a>    module_lists[[i]]<span class="ot">=</span>  module.gene.assign[module.gene.assign<span class="sc">==</span> i] <span class="sc">|&gt;</span>  <span class="fu">names</span>()</span>
<span id="cb89-36"><a href="wgcna-pipeline.html#cb89-36" tabindex="-1"></a>  }</span>
<span id="cb89-37"><a href="wgcna-pipeline.html#cb89-37" tabindex="-1"></a>  </span>
<span id="cb89-38"><a href="wgcna-pipeline.html#cb89-38" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">ldply</span>(module_lists, rbind) <span class="sc">|&gt;</span>  <span class="fu">t</span>()</span>
<span id="cb89-39"><a href="wgcna-pipeline.html#cb89-39" tabindex="-1"></a>  df[<span class="fu">is.na</span>(df)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb89-40"><a href="wgcna-pipeline.html#cb89-40" tabindex="-1"></a>  df<span class="ot">=</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb89-41"><a href="wgcna-pipeline.html#cb89-41" tabindex="-1"></a>  <span class="fu">names</span>(df) <span class="ot">=</span> df[<span class="dv">1</span>,] <span class="sc">|&gt;</span>  <span class="fu">as.vector</span>()</span>
<span id="cb89-42"><a href="wgcna-pipeline.html#cb89-42" tabindex="-1"></a>  modules<span class="ot">=</span> df[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb89-43"><a href="wgcna-pipeline.html#cb89-43" tabindex="-1"></a>  <span class="fu">write.csv</span>(modules, <span class="fu">paste0</span>(<span class="st">&quot;output/&quot;</span>, <span class="st">&quot;modules.csv&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb89-44"><a href="wgcna-pipeline.html#cb89-44" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;modules file is saved!&quot;</span>)</span>
<span id="cb89-45"><a href="wgcna-pipeline.html#cb89-45" tabindex="-1"></a>  </span>
<span id="cb89-46"><a href="wgcna-pipeline.html#cb89-46" tabindex="-1"></a>  </span>
<span id="cb89-47"><a href="wgcna-pipeline.html#cb89-47" tabindex="-1"></a>  <span class="co">#calculate gene significance (a measure that show to what degree the gene is correlated with phenotype) </span></span>
<span id="cb89-48"><a href="wgcna-pipeline.html#cb89-48" tabindex="-1"></a>  nSamples <span class="ot">&lt;-</span> <span class="fu">nrow</span>(datExpr)</span>
<span id="cb89-49"><a href="wgcna-pipeline.html#cb89-49" tabindex="-1"></a>  nGenes <span class="ot">&lt;-</span> <span class="fu">ncol</span>(datExpr)</span>
<span id="cb89-50"><a href="wgcna-pipeline.html#cb89-50" tabindex="-1"></a>  gene.signf.corr <span class="ot">&lt;-</span> <span class="fu">cor</span>(datExpr, metadata_binary, <span class="at">use =</span> <span class="st">&#39;p&#39;</span>)</span>
<span id="cb89-51"><a href="wgcna-pipeline.html#cb89-51" tabindex="-1"></a>  gene.signf.corr.pvals <span class="ot">&lt;-</span> <span class="fu">corPvalueStudent</span>(gene.signf.corr, nSamples)</span>
<span id="cb89-52"><a href="wgcna-pipeline.html#cb89-52" tabindex="-1"></a>  <span class="fu">write.csv</span>(gene.signf.corr, <span class="st">&quot;output/gene.trait.corr.csv&quot;</span>)</span>
<span id="cb89-53"><a href="wgcna-pipeline.html#cb89-53" tabindex="-1"></a>  gene.signf.corr<span class="ot">=</span> gene.signf.corr  <span class="sc">|&gt;</span>  <span class="fu">as.data.frame</span>()</span>
<span id="cb89-54"><a href="wgcna-pipeline.html#cb89-54" tabindex="-1"></a>  <span class="co">#Module membership (reflects intra-modular connectivity)</span></span>
<span id="cb89-55"><a href="wgcna-pipeline.html#cb89-55" tabindex="-1"></a>  <span class="co"># calculate the module membership values node i. The module membership kME(q) specifies how close node i is to module q. </span></span>
<span id="cb89-56"><a href="wgcna-pipeline.html#cb89-56" tabindex="-1"></a>  <span class="co">#KME (module membership) defined as correlation between gene expression profile and the module eigengene of certain module </span></span>
<span id="cb89-57"><a href="wgcna-pipeline.html#cb89-57" tabindex="-1"></a>  <span class="co"># if MMblue(i) is close to 1 or -1, it is highly connected to the blue module genes.</span></span>
<span id="cb89-58"><a href="wgcna-pipeline.html#cb89-58" tabindex="-1"></a>  <span class="co"># The sign of module membership encodes whether the gene has a positive or a negative relationship with the module eigengene. </span></span>
<span id="cb89-59"><a href="wgcna-pipeline.html#cb89-59" tabindex="-1"></a>  datKME <span class="ot">=</span> <span class="fu">signedKME</span>(datExpr, module_eigengenes)</span>
<span id="cb89-60"><a href="wgcna-pipeline.html#cb89-60" tabindex="-1"></a>  <span class="fu">write.csv</span>(datKME, <span class="st">&quot;output/module_membership.csv&quot;</span>)</span>
<span id="cb89-61"><a href="wgcna-pipeline.html#cb89-61" tabindex="-1"></a>  </span>
<span id="cb89-62"><a href="wgcna-pipeline.html#cb89-62" tabindex="-1"></a>  </span>
<span id="cb89-63"><a href="wgcna-pipeline.html#cb89-63" tabindex="-1"></a>  <span class="co">#extract hub genes</span></span>
<span id="cb89-64"><a href="wgcna-pipeline.html#cb89-64" tabindex="-1"></a>  <span class="co">#Extract hubs for each modules</span></span>
<span id="cb89-65"><a href="wgcna-pipeline.html#cb89-65" tabindex="-1"></a>  module_names<span class="ot">=</span> <span class="fu">names</span>(module_eigengenes)</span>
<span id="cb89-66"><a href="wgcna-pipeline.html#cb89-66" tabindex="-1"></a>  hub_lists <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>, <span class="fu">length</span>(module_names))</span>
<span id="cb89-67"><a href="wgcna-pipeline.html#cb89-67" tabindex="-1"></a>  <span class="fu">names</span>(hub_lists)<span class="ot">=</span> <span class="fu">names</span>(module_eigengenes)</span>
<span id="cb89-68"><a href="wgcna-pipeline.html#cb89-68" tabindex="-1"></a>  threshold<span class="ot">=</span> hub_threshold</span>
<span id="cb89-69"><a href="wgcna-pipeline.html#cb89-69" tabindex="-1"></a>  </span>
<span id="cb89-70"><a href="wgcna-pipeline.html#cb89-70" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(module_names)){</span>
<span id="cb89-71"><a href="wgcna-pipeline.html#cb89-71" tabindex="-1"></a>    hub_mod<span class="ot">=</span> <span class="fu">row.names</span>(datKME)[datKME[,j] <span class="sc">&gt;</span> threshold]</span>
<span id="cb89-72"><a href="wgcna-pipeline.html#cb89-72" tabindex="-1"></a>    hub_lists[[j]]<span class="ot">=</span> <span class="fu">intersect</span>(<span class="fu">unlist</span>(modules[j]),  hub_mod)</span>
<span id="cb89-73"><a href="wgcna-pipeline.html#cb89-73" tabindex="-1"></a>  }</span>
<span id="cb89-74"><a href="wgcna-pipeline.html#cb89-74" tabindex="-1"></a>  </span>
<span id="cb89-75"><a href="wgcna-pipeline.html#cb89-75" tabindex="-1"></a>  df<span class="ot">=</span> <span class="fu">ldply</span>(hub_lists, rbind) <span class="sc">|&gt;</span>  <span class="fu">t</span>() </span>
<span id="cb89-76"><a href="wgcna-pipeline.html#cb89-76" tabindex="-1"></a>  df[<span class="fu">is.na</span>(df)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb89-77"><a href="wgcna-pipeline.html#cb89-77" tabindex="-1"></a>  df<span class="ot">=</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb89-78"><a href="wgcna-pipeline.html#cb89-78" tabindex="-1"></a>  <span class="fu">names</span>(df) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">unlist</span>(df[<span class="dv">1</span>,]), <span class="st">&quot;_hub&quot;</span>) </span>
<span id="cb89-79"><a href="wgcna-pipeline.html#cb89-79" tabindex="-1"></a>  hubs<span class="ot">=</span> df[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb89-80"><a href="wgcna-pipeline.html#cb89-80" tabindex="-1"></a>  <span class="fu">write.csv</span>(hubs, <span class="fu">paste0</span>(<span class="st">&quot;output/&quot;</span>, <span class="st">&quot;module_hubs.csv&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb89-81"><a href="wgcna-pipeline.html#cb89-81" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;hubs file is saved!&quot;</span>)</span>
<span id="cb89-82"><a href="wgcna-pipeline.html#cb89-82" tabindex="-1"></a>  </span>
<span id="cb89-83"><a href="wgcna-pipeline.html#cb89-83" tabindex="-1"></a>  <span class="co">#Extract hubs correlated with phenotype</span></span>
<span id="cb89-84"><a href="wgcna-pipeline.html#cb89-84" tabindex="-1"></a>  hub_pheno <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>, <span class="at">length=</span> <span class="fu">ncol</span>(gene.signf.corr))</span>
<span id="cb89-85"><a href="wgcna-pipeline.html#cb89-85" tabindex="-1"></a>  <span class="fu">names</span>(hub_pheno)<span class="ot">=</span> <span class="fu">names</span>(gene.signf.corr)</span>
<span id="cb89-86"><a href="wgcna-pipeline.html#cb89-86" tabindex="-1"></a>  GS_threshold<span class="ot">=</span> GS_threshold</span>
<span id="cb89-87"><a href="wgcna-pipeline.html#cb89-87" tabindex="-1"></a>  </span>
<span id="cb89-88"><a href="wgcna-pipeline.html#cb89-88" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(gene.signf.corr)){</span>
<span id="cb89-89"><a href="wgcna-pipeline.html#cb89-89" tabindex="-1"></a>    </span>
<span id="cb89-90"><a href="wgcna-pipeline.html#cb89-90" tabindex="-1"></a>    all_hubs<span class="ot">=</span> hub_lists <span class="sc">|&gt;</span>  <span class="fu">unlist</span>() </span>
<span id="cb89-91"><a href="wgcna-pipeline.html#cb89-91" tabindex="-1"></a>    all_hubs<span class="ot">=</span> all_hubs[all_hubs <span class="sc">!=</span><span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(all_hubs)]</span>
<span id="cb89-92"><a href="wgcna-pipeline.html#cb89-92" tabindex="-1"></a>    pheno_h<span class="ot">=</span> <span class="fu">row.names</span>(gene.signf.corr)[gene.signf.corr[,j] <span class="sc">&gt;</span> GS_threshold]</span>
<span id="cb89-93"><a href="wgcna-pipeline.html#cb89-93" tabindex="-1"></a>    pheno_h<span class="ot">=</span> pheno_h[pheno_h <span class="sc">%in%</span> all_hubs]</span>
<span id="cb89-94"><a href="wgcna-pipeline.html#cb89-94" tabindex="-1"></a>    hub_pheno[[j]] <span class="ot">=</span> pheno_h</span>
<span id="cb89-95"><a href="wgcna-pipeline.html#cb89-95" tabindex="-1"></a>  }</span>
<span id="cb89-96"><a href="wgcna-pipeline.html#cb89-96" tabindex="-1"></a>  df<span class="ot">=</span> <span class="fu">ldply</span>(hub_pheno, rbind) <span class="sc">|&gt;</span>  <span class="fu">t</span>() </span>
<span id="cb89-97"><a href="wgcna-pipeline.html#cb89-97" tabindex="-1"></a>  df[<span class="fu">is.na</span>(df)] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb89-98"><a href="wgcna-pipeline.html#cb89-98" tabindex="-1"></a>  df<span class="ot">=</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb89-99"><a href="wgcna-pipeline.html#cb89-99" tabindex="-1"></a>  <span class="fu">names</span>(df) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">unlist</span>(df[<span class="dv">1</span>,]), <span class="st">&quot;_hub&quot;</span>) </span>
<span id="cb89-100"><a href="wgcna-pipeline.html#cb89-100" tabindex="-1"></a>  hubs<span class="ot">=</span> df[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb89-101"><a href="wgcna-pipeline.html#cb89-101" tabindex="-1"></a>  <span class="co">#write.csv(hubs, paste0(&quot;output/&quot;, &quot;phenotype_cor_hubs.csv&quot;), row.names = FALSE)</span></span>
<span id="cb89-102"><a href="wgcna-pipeline.html#cb89-102" tabindex="-1"></a>  <span class="co">#print(&quot;hubs_pheno file is saved!&quot;)</span></span>
<span id="cb89-103"><a href="wgcna-pipeline.html#cb89-103" tabindex="-1"></a>  </span>
<span id="cb89-104"><a href="wgcna-pipeline.html#cb89-104" tabindex="-1"></a></span>
<span id="cb89-105"><a href="wgcna-pipeline.html#cb89-105" tabindex="-1"></a>  <span class="co">#export networks for cytoscape</span></span>
<span id="cb89-106"><a href="wgcna-pipeline.html#cb89-106" tabindex="-1"></a>  module_names<span class="ot">=</span> <span class="fu">names</span>(module_eigengenes)</span>
<span id="cb89-107"><a href="wgcna-pipeline.html#cb89-107" tabindex="-1"></a>  <span class="cf">for</span>( i <span class="cf">in</span>   module_names) {</span>
<span id="cb89-108"><a href="wgcna-pipeline.html#cb89-108" tabindex="-1"></a>    datexpr_mod <span class="ot">=</span> datExpr[,module.gene.assign <span class="sc">==</span> i]</span>
<span id="cb89-109"><a href="wgcna-pipeline.html#cb89-109" tabindex="-1"></a>    TOM_mod <span class="ot">=</span> <span class="fu">TOMsimilarityFromExpr</span>(datexpr_mod, <span class="at">power =</span> power, <span class="at">networkType =</span> <span class="st">&quot;signed&quot;</span>, <span class="at">TOMType=</span><span class="st">&quot;signed Nowick&quot;</span>);</span>
<span id="cb89-110"><a href="wgcna-pipeline.html#cb89-110" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">colnames</span>(datexpr_mod)</span>
<span id="cb89-111"><a href="wgcna-pipeline.html#cb89-111" tabindex="-1"></a>    <span class="fu">dimnames</span>(TOM_mod) <span class="ot">=</span> <span class="fu">list</span>(genes, genes)</span>
<span id="cb89-112"><a href="wgcna-pipeline.html#cb89-112" tabindex="-1"></a>    <span class="co"># nTop = 30;</span></span>
<span id="cb89-113"><a href="wgcna-pipeline.html#cb89-113" tabindex="-1"></a>    <span class="co"># IMConn = softConnectivity(datExpr[, probes]);</span></span>
<span id="cb89-114"><a href="wgcna-pipeline.html#cb89-114" tabindex="-1"></a>    <span class="co"># top = (rank(-IMConn) &lt;= nTop)</span></span>
<span id="cb89-115"><a href="wgcna-pipeline.html#cb89-115" tabindex="-1"></a>    cyt <span class="ot">=</span> <span class="fu">exportNetworkToCytoscape</span>(TOM_mod,</span>
<span id="cb89-116"><a href="wgcna-pipeline.html#cb89-116" tabindex="-1"></a>                                   <span class="at">edgeFile =</span> <span class="fu">paste</span>(<span class="st">&quot;output/CytoscapeInput-edges-.02tom&quot;</span>, <span class="fu">paste</span>(i, <span class="at">collapse=</span><span class="st">&quot;-&quot;</span>), <span class="st">&quot;.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb89-117"><a href="wgcna-pipeline.html#cb89-117" tabindex="-1"></a>                                   <span class="co">#nodeFile = paste(&quot;output/CytoscapeInput-nodes-&quot;, paste(i, collapse=&quot;-&quot;), &quot;.txt&quot;, sep=&quot;&quot;),</span></span>
<span id="cb89-118"><a href="wgcna-pipeline.html#cb89-118" tabindex="-1"></a>                                   <span class="at">weighted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb89-119"><a href="wgcna-pipeline.html#cb89-119" tabindex="-1"></a>                                   <span class="at">threshold =</span> <span class="fl">0.02</span>, <span class="co">#threshold for including edges in the output, Default is 0.02</span></span>
<span id="cb89-120"><a href="wgcna-pipeline.html#cb89-120" tabindex="-1"></a>                                   <span class="at">nodeNames =</span> genes,</span>
<span id="cb89-121"><a href="wgcna-pipeline.html#cb89-121" tabindex="-1"></a>                                   <span class="at">altNodeNames =</span> genes);</span>
<span id="cb89-122"><a href="wgcna-pipeline.html#cb89-122" tabindex="-1"></a>  }</span>
<span id="cb89-123"><a href="wgcna-pipeline.html#cb89-123" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;networks exported!&quot;</span>)</span>
<span id="cb89-124"><a href="wgcna-pipeline.html#cb89-124" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;ALL iS DONE!&quot;</span>)</span>
<span id="cb89-125"><a href="wgcna-pipeline.html#cb89-125" tabindex="-1"></a>  <span class="fu">return</span>(net)</span>
<span id="cb89-126"><a href="wgcna-pipeline.html#cb89-126" tabindex="-1"></a>  </span>
<span id="cb89-127"><a href="wgcna-pipeline.html#cb89-127" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="run-wgcna" class="section level3 hasAnchor" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> <strong>Run WGCNA</strong><a href="wgcna-pipeline.html#run-wgcna" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="wgcna-pipeline.html#cb90-1" tabindex="-1"></a>res<span class="ot">=</span> <span class="fu">Run_WGCNA</span>(data, <span class="at">metadata_binary=</span>metadata_binary, <span class="at">power=</span>power, <span class="at">minModuleSize=</span> minModuleSize,</span>
<span id="cb90-2"><a href="wgcna-pipeline.html#cb90-2" tabindex="-1"></a>               <span class="at">hub_threshold=</span> hub_threshold, <span class="at">GS_threshold=</span>GS_threshold, <span class="at">mapping_file=</span><span class="cn">NULL</span>)</span></code></pre></div>
<pre><code>##  Flagging genes and samples with too many missing values...
##   ..step 1
## [1] &quot;module detection is done&quot;
## [1] &quot;module_eigengenes file is saved&quot;
## [1] &quot;gene_module_assignment file is saved&quot;
## [1] &quot;modules file is saved!&quot;
## [1] &quot;hubs file is saved!&quot;
## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.
## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.
## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.
## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.
## [1] &quot;networks exported!&quot;
## [1] &quot;ALL iS DONE!&quot;</code></pre>
<pre><code>## R version 4.4.1 (2024-06-14 ucrt)
## Platform: x86_64-w64-mingw32/x64
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## 
## locale:
## [1] LC_COLLATE=English_United States.utf8 
## [2] LC_CTYPE=English_United States.utf8   
## [3] LC_MONETARY=English_United States.utf8
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.utf8    
## 
## time zone: Africa/Cairo
## tzcode source: internal
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] WGCNA_1.72-5          fastcluster_1.2.6     dynamicTreeCut_1.63-1
##  [4] ggrepel_0.9.6         viridis_0.6.5         fields_16.2          
##  [7] viridisLite_0.4.2     spam_2.10-0           biomaRt_2.61.2       
## [10] ComplexHeatmap_2.21.0 circlize_0.4.16       RColorBrewer_1.1-3   
## [13] memoise_2.0.1         caret_6.0-94          lattice_0.22-6       
## [16] pls_2.8-3             Rserve_1.8-13         MetaboAnalystR_3.2.0 
## [19] cowplot_1.1.3         DT_0.33               openxlsx_4.2.6.1     
## [22] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.1        
## [25] purrr_1.0.2           readr_2.1.5           tidyr_1.3.1          
## [28] ggplot2_3.5.1         tidyverse_2.0.0       dplyr_1.1.4          
## [31] plyr_1.8.9            tibble_3.2.1         
## 
## loaded via a namespace (and not attached):
##   [1] splines_4.4.1           filelock_1.0.3          bitops_1.0-7           
##   [4] preprocessCore_1.67.0   hardhat_1.4.0           pROC_1.18.5            
##   [7] rpart_4.1.23            httr2_1.0.2             lifecycle_1.0.4        
##  [10] edgeR_4.3.5             doParallel_1.0.17       globals_0.16.3         
##  [13] MASS_7.3-60.2           scrime_1.3.5            crosstalk_1.2.1        
##  [16] backports_1.5.0         magrittr_2.0.3          Hmisc_5.1-3            
##  [19] limma_3.61.5            plotly_4.10.4           sass_0.4.9             
##  [22] rmarkdown_2.27          jquerylib_0.1.4         yaml_2.3.10            
##  [25] zip_2.3.1               DBI_1.2.3               maps_3.4.2             
##  [28] zlibbioc_1.50.0         BiocGenerics_0.52.0     nnet_7.3-19            
##  [31] rappdirs_0.3.3          ipred_0.9-15            GenomeInfoDbData_1.2.12
##  [34] lava_1.8.0              IRanges_2.38.1          S4Vectors_0.42.1       
##  [37] listenv_0.9.1           ellipse_0.5.0           parallelly_1.38.0      
##  [40] codetools_0.2-20        xml2_1.3.6              RApiSerialize_0.1.3    
##  [43] tidyselect_1.2.1        shape_1.4.6.1           UCSC.utils_1.1.0       
##  [46] farver_2.1.2            base64enc_0.1-3         BiocFileCache_2.13.0   
##  [49] matrixStats_1.3.0       stats4_4.4.1            jsonlite_1.8.8         
##  [52] GetoptLong_1.0.5        multtest_2.61.0         e1071_1.7-14           
##  [55] Formula_1.2-5           survival_3.6-4          iterators_1.0.14       
##  [58] systemfonts_1.2.3       foreach_1.5.2           progress_1.2.3         
##  [61] tools_4.4.1             ragg_1.3.2              Rcpp_1.0.13            
##  [64] glue_1.7.0              gridExtra_2.3           prodlim_2024.06.25     
##  [67] xfun_0.46               GenomeInfoDb_1.41.1     crmn_0.0.21            
##  [70] withr_3.0.0             fastmap_1.2.0           caTools_1.18.2         
##  [73] digest_0.6.36           timechange_0.3.0        R6_2.5.1               
##  [76] textshaping_0.4.0       colorspace_2.1-1        GO.db_3.19.1           
##  [79] Cairo_1.6-2             gtools_3.9.5            RSQLite_2.3.7          
##  [82] generics_0.1.3          data.table_1.15.4       recipes_1.1.0          
##  [85] class_7.3-22            prettyunits_1.2.0       httr_1.4.7             
##  [88] htmlwidgets_1.6.4       ModelMetrics_1.2.2.2    pkgconfig_2.0.3        
##  [91] gtable_0.3.5            timeDate_4032.109       blob_1.2.4             
##  [94] siggenes_1.79.0         impute_1.79.0           XVector_0.44.0         
##  [97] htmltools_0.5.8.1       dotCall64_1.1-1         bookdown_0.40          
## [100] fgsea_1.31.0            clue_0.3-65             scales_1.3.0           
## [103] Biobase_2.64.0          png_0.1-8               gower_1.0.1            
## [106] knitr_1.48              rstudioapi_0.16.0       tzdb_0.4.0             
## [109] reshape2_1.4.4          rjson_0.2.21            checkmate_2.3.1        
## [112] curl_5.2.1              nlme_3.1-164            proxy_0.4-27           
## [115] cachem_1.1.0            GlobalOptions_0.1.2     KernSmooth_2.23-24     
## [118] parallel_4.4.1          foreign_0.8-86          AnnotationDbi_1.67.0   
## [121] pillar_1.11.0           vctrs_0.6.5             gplots_3.1.3.1         
## [124] pcaMethods_1.97.0       stringfish_0.16.0       dbplyr_2.5.0           
## [127] cluster_2.1.6           htmlTable_2.4.3         evaluate_0.24.0        
## [130] magick_2.8.4            cli_3.6.3               locfit_1.5-9.10        
## [133] compiler_4.4.1          rlang_1.1.4             crayon_1.5.3           
## [136] future.apply_1.11.2     labeling_0.4.3          stringi_1.8.4          
## [139] BiocParallel_1.39.0     munsell_0.5.1           Biostrings_2.72.1      
## [142] lazyeval_0.2.2          Matrix_1.7-0            hms_1.1.3              
## [145] glasso_1.11             bit64_4.0.5             future_1.33.2          
## [148] KEGGREST_1.45.1         statmod_1.5.0           highr_0.11             
## [151] qs_0.26.3               igraph_2.0.3            RcppParallel_5.1.8     
## [154] bslib_0.8.0             fastmatch_1.1-4         bit_4.0.5</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="functional-enrichment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wgcna-visualizations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/Menna-Arafat/CRC_project/5_Automated_WGCNA_main.rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["mybook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
