<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 7 WGCNA Visualizations | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</title>
  <meta name="description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 7 WGCNA Visualizations | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  <meta name="github-repo" content="https://github.com/Menna-Arafat/CRC_project" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 7 WGCNA Visualizations | Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows" />
  
  <meta name="twitter:description" content="This book presents an end-to-end pipeline for metabolomics analysis, including raw data preprocessing, downstream analysis, visualization, Co-expression Network Analysis (WGCNA), and MetaboAnalyst integration. The workflow does not include the manual curation step that is to compare MS/MS experimental spectra to reference libraries (e.g., HMDB, METLIN, GNPS, MassBank). The whole pipeline was applied in our published study: Ramzy, A., Abdelmoneim, T.K., Arafat, M. et al. (2025), “Metabolomic analysis reveals key changes in amino acid metabolism in colorectal cancer patients” (Amino Acids, 57:22). https://doi.org/10.1007/s00726-025-03448-3." />
  

<meta name="author" content="Menna Arafat" />


<meta name="date" content="2025-08-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="wgcna-pipeline.html"/>
<link rel="next" href="module-eigengene-differentiation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network-Based Metabolomics Analysis in R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html"><i class="fa fa-check"></i><b>1</b> Metabolomic Raw data Processing</a>
<ul>
<li class="chapter" data-level="1.1" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#internal-functions"><i class="fa fa-check"></i><b>1.1</b> <strong>Internal Functions</strong></a></li>
<li class="chapter" data-level="1.2" data-path="metabolomic-raw-data-processing.html"><a href="metabolomic-raw-data-processing.html#wrapper-function"><i class="fa fa-check"></i><b>1.2</b> <strong>Wrapper Function</strong></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html"><i class="fa fa-check"></i><b>2</b> Dimensionality Reduction Techniques</a>
<ul>
<li class="chapter" data-level="2.1" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#input-data-preparation-for-metaboanalyst"><i class="fa fa-check"></i><b>2.1</b> <strong>Input data Preparation for Metaboanalyst</strong></a></li>
<li class="chapter" data-level="2.2" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pca"><i class="fa fa-check"></i><b>2.2</b> <strong>PCA</strong></a></li>
<li class="chapter" data-level="2.3" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#pls-da"><i class="fa fa-check"></i><b>2.3</b> <strong>PLS-DA</strong></a></li>
<li class="chapter" data-level="2.4" data-path="dimensionality-reduction-techniques.html"><a href="dimensionality-reduction-techniques.html#splsda"><i class="fa fa-check"></i><b>2.4</b> <strong>splsda</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html"><i class="fa fa-check"></i><b>3</b> Metaboanalyst Pipeline</a>
<ul>
<li class="chapter" data-level="3.1" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#normality-assessment"><i class="fa fa-check"></i><b>3.1</b> <strong>Normality Assessment</strong></a></li>
<li class="chapter" data-level="3.2" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#metaboanalystr-object-initialization"><i class="fa fa-check"></i><b>3.2</b> <strong>MetaboAnalystR Object Initialization</strong></a></li>
<li class="chapter" data-level="3.3" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#auto-scaleing-z-score-normalization"><i class="fa fa-check"></i><b>3.3</b> <strong>Auto-scaleing (z-score Normalization)</strong></a></li>
<li class="chapter" data-level="3.4" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#post-normalization-normality-check"><i class="fa fa-check"></i><b>3.4</b> <strong>Post-Normalization Normality Check</strong></a></li>
<li class="chapter" data-level="3.5" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.5</b> <strong>Fold change Analysis</strong></a></li>
<li class="chapter" data-level="3.6" data-path="metaboanalyst-pipeline.html"><a href="metaboanalyst-pipeline.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.6</b> <strong>Differential Expression Analysis</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html"><i class="fa fa-check"></i><b>4</b> Downstream Visualizations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#volcano-plot"><i class="fa fa-check"></i><b>4.1</b> <strong>Volcano Plot</strong></a></li>
<li class="chapter" data-level="4.2" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#heatmap-plot"><i class="fa fa-check"></i><b>4.2</b> <strong>Heatmap Plot</strong></a></li>
<li class="chapter" data-level="4.3" data-path="downstream-visualizations.html"><a href="downstream-visualizations.html#box-plots"><i class="fa fa-check"></i><b>4.3</b> <strong>Box Plots</strong></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment.html"><a href="functional-enrichment.html"><i class="fa fa-check"></i><b>5</b> Functional Enrichment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment.html"><a href="functional-enrichment.html#data-preparation"><i class="fa fa-check"></i><b>5.1</b> <strong>Data Preparation</strong></a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment.html"><a href="functional-enrichment.html#enrichment-plot"><i class="fa fa-check"></i><b>5.2</b> <strong>Enrichment plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html"><i class="fa fa-check"></i><b>6</b> WGCNA Pipeline</a>
<ul>
<li class="chapter" data-level="6.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-pipeline-1"><i class="fa fa-check"></i><b>6.1</b> <strong>WGCNA Pipeline</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#choose-appropriate-power-for-soft-thresholding"><i class="fa fa-check"></i><b>6.1.1</b> <strong>Choose appropriate power for soft-thresholding</strong></a></li>
<li class="chapter" data-level="6.1.2" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#set-main-parameters"><i class="fa fa-check"></i><b>6.1.2</b> <strong>Set Main Parameters</strong></a></li>
<li class="chapter" data-level="6.1.3" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#wgcna-main-function"><i class="fa fa-check"></i><b>6.1.3</b> <strong>WGCNA Main Function</strong></a></li>
<li class="chapter" data-level="6.1.4" data-path="wgcna-pipeline.html"><a href="wgcna-pipeline.html#run-wgcna"><i class="fa fa-check"></i><b>6.1.4</b> <strong>Run WGCNA</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html"><i class="fa fa-check"></i><b>7</b> WGCNA Visualizations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-phenotype-correlation-plot"><i class="fa fa-check"></i><b>7.1</b> <strong>Module-phenotype Correlation Plot</strong></a></li>
<li class="chapter" data-level="7.2" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#samples-dendrogram-plot"><i class="fa fa-check"></i><b>7.2</b> <strong>Samples Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.3" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#cluster-dendrogram-plot"><i class="fa fa-check"></i><b>7.3</b> <strong>Cluster Dendrogram Plot</strong></a></li>
<li class="chapter" data-level="7.4" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-heatmap-of-correlated-features-across-all-modules"><i class="fa fa-check"></i><b>7.4</b> <strong>TOM plot (Heatmap of Correlated Features Across All Modules)</strong></a></li>
<li class="chapter" data-level="7.5" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#mds-plot"><i class="fa fa-check"></i><b>7.5</b> <strong>MDS plot</strong></a></li>
<li class="chapter" data-level="7.6" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#tom-plot-for-selected-features"><i class="fa fa-check"></i><b>7.6</b> <strong>TOM plot for Selected Features</strong></a></li>
<li class="chapter" data-level="7.7" data-path="wgcna-visualizations.html"><a href="wgcna-visualizations.html#module-membership-vs-gene-significance-scatter-plot"><i class="fa fa-check"></i><b>7.7</b> <strong>Module Membership Vs Gene Significance Scatter plot</strong></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html"><i class="fa fa-check"></i><b>8</b> Module Eigengene Differentiation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#function-for-module-eigengene-differentiation"><i class="fa fa-check"></i><b>8.1</b> <strong>Function for Module eigengene differentiation</strong></a></li>
<li class="chapter" data-level="8.2" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#horizontal-bar-plot"><i class="fa fa-check"></i><b>8.2</b> <strong>Horizontal Bar plot</strong></a></li>
<li class="chapter" data-level="8.3" data-path="module-eigengene-differentiation.html"><a href="module-eigengene-differentiation.html#chord-plot-of-hubgenes-and-corresponding-modules"><i class="fa fa-check"></i><b>8.3</b> <strong>Chord plot of hubgenes and corresponding modules</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network-Based Metabolomics in R: Data Preprocessing, Co-expression Network Analysis, and MetaboAnalyst Workflows</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wgcna-visualizations" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Section 7</span> WGCNA Visualizations<a href="wgcna-visualizations.html#wgcna-visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p><strong>Description</strong>: This pipeline performs all visualizations for the WGCNA pipeline.</p>
</blockquote>
<p><strong>Project Initialization</strong></p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="wgcna-visualizations.html#cb93-1" tabindex="-1"></a><span class="co">#Sets the working directory and creates subfolders for organizing outputs.</span></span>
<span id="cb93-2"><a href="wgcna-visualizations.html#cb93-2" tabindex="-1"></a>mypath<span class="ot">=</span> <span class="st">&quot;C:/Users/USER/Documents/Github/CRC_project/&quot;</span></span>
<span id="cb93-3"><a href="wgcna-visualizations.html#cb93-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;output&quot;</span>)</span>
<span id="cb93-4"><a href="wgcna-visualizations.html#cb93-4" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;plots&quot;</span>)</span>
<span id="cb93-5"><a href="wgcna-visualizations.html#cb93-5" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;input&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="wgcna-visualizations.html#cb94-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb94-2"><a href="wgcna-visualizations.html#cb94-2" tabindex="-1"></a><span class="fu">library</span>(gtools)</span>
<span id="cb94-3"><a href="wgcna-visualizations.html#cb94-3" tabindex="-1"></a><span class="fu">library</span>(pROC)</span>
<span id="cb94-4"><a href="wgcna-visualizations.html#cb94-4" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb94-5"><a href="wgcna-visualizations.html#cb94-5" tabindex="-1"></a><span class="fu">library</span>(ggdendro)</span>
<span id="cb94-6"><a href="wgcna-visualizations.html#cb94-6" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb94-7"><a href="wgcna-visualizations.html#cb94-7" tabindex="-1"></a><span class="fu">library</span>(stats)</span>
<span id="cb94-8"><a href="wgcna-visualizations.html#cb94-8" tabindex="-1"></a><span class="fu">library</span>(flashClust)</span>
<span id="cb94-9"><a href="wgcna-visualizations.html#cb94-9" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb94-10"><a href="wgcna-visualizations.html#cb94-10" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb94-11"><a href="wgcna-visualizations.html#cb94-11" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb94-12"><a href="wgcna-visualizations.html#cb94-12" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb94-13"><a href="wgcna-visualizations.html#cb94-13" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb94-14"><a href="wgcna-visualizations.html#cb94-14" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb94-15"><a href="wgcna-visualizations.html#cb94-15" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb94-16"><a href="wgcna-visualizations.html#cb94-16" tabindex="-1"></a><span class="fu">library</span>(gplots)</span>
<span id="cb94-17"><a href="wgcna-visualizations.html#cb94-17" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb94-18"><a href="wgcna-visualizations.html#cb94-18" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb94-19"><a href="wgcna-visualizations.html#cb94-19" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb94-20"><a href="wgcna-visualizations.html#cb94-20" tabindex="-1"></a><span class="fu">allowWGCNAThreads</span>()         </span></code></pre></div>
<pre><code>## Allowing multi-threading with up to 4 threads.</code></pre>
<p><strong>Prepare Data</strong></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="wgcna-visualizations.html#cb96-1" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb96-2"><a href="wgcna-visualizations.html#cb96-2" tabindex="-1"></a>dir<span class="ot">=</span> <span class="fu">paste0</span>(mypath, <span class="st">&quot;output/&quot;</span>)</span>
<span id="cb96-3"><a href="wgcna-visualizations.html#cb96-3" tabindex="-1"></a><span class="co"># expression data</span></span>
<span id="cb96-4"><a href="wgcna-visualizations.html#cb96-4" tabindex="-1"></a>data<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;input/data_for_downstream.csv&quot;</span>)) </span>
<span id="cb96-5"><a href="wgcna-visualizations.html#cb96-5" tabindex="-1"></a>data <span class="ot">=</span> data <span class="sc">|&gt;</span>  <span class="fu">column_to_rownames</span>(<span class="fu">colnames</span>(data)[<span class="dv">1</span>]) </span>
<span id="cb96-6"><a href="wgcna-visualizations.html#cb96-6" tabindex="-1"></a><span class="co"># differential expression file</span></span>
<span id="cb96-7"><a href="wgcna-visualizations.html#cb96-7" tabindex="-1"></a>de<span class="ot">=</span>   <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath, <span class="st">&quot;output/DE_sig.csv&quot;</span>))</span>
<span id="cb96-8"><a href="wgcna-visualizations.html#cb96-8" tabindex="-1"></a>de<span class="ot">=</span>de <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;=</span> <span class="fu">log2</span>(<span class="dv">3</span>))</span>
<span id="cb96-9"><a href="wgcna-visualizations.html#cb96-9" tabindex="-1"></a>module_eigengenes<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/module_eigengenes.csv&quot;</span>)) <span class="sc">|&gt;</span>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;X&quot;</span>)</span>
<span id="cb96-10"><a href="wgcna-visualizations.html#cb96-10" tabindex="-1"></a></span>
<span id="cb96-11"><a href="wgcna-visualizations.html#cb96-11" tabindex="-1"></a></span>
<span id="cb96-12"><a href="wgcna-visualizations.html#cb96-12" tabindex="-1"></a>group_dist<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(data))</span>
<span id="cb96-13"><a href="wgcna-visualizations.html#cb96-13" tabindex="-1"></a>group_levels<span class="ot">=</span> <span class="fu">unique</span>(group_dist)</span>
<span id="cb96-14"><a href="wgcna-visualizations.html#cb96-14" tabindex="-1"></a></span>
<span id="cb96-15"><a href="wgcna-visualizations.html#cb96-15" tabindex="-1"></a>group_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#fc8d62&quot;</span>, <span class="st">&quot;#66c2a5&quot;</span>)</span>
<span id="cb96-16"><a href="wgcna-visualizations.html#cb96-16" tabindex="-1"></a><span class="fu">names</span>(group_colors) <span class="ot">&lt;-</span> group_levels</span>
<span id="cb96-17"><a href="wgcna-visualizations.html#cb96-17" tabindex="-1"></a></span>
<span id="cb96-18"><a href="wgcna-visualizations.html#cb96-18" tabindex="-1"></a><span class="co">#change type to numeric</span></span>
<span id="cb96-19"><a href="wgcna-visualizations.html#cb96-19" tabindex="-1"></a>data[]<span class="ot">=</span> <span class="fu">lapply</span>(data, as.numeric)</span>
<span id="cb96-20"><a href="wgcna-visualizations.html#cb96-20" tabindex="-1"></a>datExpr<span class="ot">=</span> <span class="fu">t</span>(data) <span class="co">#samples become in row</span></span>
<span id="cb96-21"><a href="wgcna-visualizations.html#cb96-21" tabindex="-1"></a><span class="co">#This function checks data for missing entries, entries with weights below a threshold, and zero-variance genes, </span></span>
<span id="cb96-22"><a href="wgcna-visualizations.html#cb96-22" tabindex="-1"></a>goods <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##  Flagging genes and samples with too many missing values...
##   ..step 1</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="wgcna-visualizations.html#cb98-1" tabindex="-1"></a>datExpr<span class="ot">=</span> datExpr[goods<span class="sc">$</span>goodSamples<span class="sc">==</span> <span class="cn">TRUE</span>, goods<span class="sc">$</span>goodGenes <span class="sc">==</span> <span class="cn">TRUE</span> ]</span></code></pre></div>
<p><strong>Prepare Metadata</strong></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="wgcna-visualizations.html#cb99-1" tabindex="-1"></a><span class="co">#metadata</span></span>
<span id="cb99-2"><a href="wgcna-visualizations.html#cb99-2" tabindex="-1"></a>metadata<span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample=</span> <span class="fu">colnames</span>(data) ,</span>
<span id="cb99-3"><a href="wgcna-visualizations.html#cb99-3" tabindex="-1"></a>                     <span class="at">condition=</span> group_dist,</span>
<span id="cb99-4"><a href="wgcna-visualizations.html#cb99-4" tabindex="-1"></a>                     <span class="at">cond_binary=</span> <span class="fu">as.numeric</span>(<span class="fu">binarizeCategoricalColumns</span>(group_dist,</span>
<span id="cb99-5"><a href="wgcna-visualizations.html#cb99-5" tabindex="-1"></a>                                                             <span class="at">levelOrder=</span> <span class="fu">list</span>(group_levels[<span class="dv">2</span>], group_levels[<span class="dv">1</span>]))[[<span class="dv">1</span>]]))</span>
<span id="cb99-6"><a href="wgcna-visualizations.html#cb99-6" tabindex="-1"></a></span>
<span id="cb99-7"><a href="wgcna-visualizations.html#cb99-7" tabindex="-1"></a><span class="co">#metadata= read.csv(&quot;input/metadata_test.csv&quot; )</span></span>
<span id="cb99-8"><a href="wgcna-visualizations.html#cb99-8" tabindex="-1"></a>metadata <span class="ot">=</span> metadata <span class="sc">|&gt;</span>   <span class="fu">column_to_rownames</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb99-9"><a href="wgcna-visualizations.html#cb99-9" tabindex="-1"></a>metadata<span class="sc">$</span>condition<span class="ot">=</span> <span class="fu">factor</span>(metadata<span class="sc">$</span>condition, <span class="at">levels =</span> group_levels)</span>
<span id="cb99-10"><a href="wgcna-visualizations.html#cb99-10" tabindex="-1"></a>design<span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span>condition , metadata)</span>
<span id="cb99-11"><a href="wgcna-visualizations.html#cb99-11" tabindex="-1"></a><span class="fu">head</span>(design)</span></code></pre></div>
<pre><code>##        conditionCRC conditionCtrl
## CRC_01            1             0
## CRC_02            1             0
## CRC_03            1             0
## CRC_04            1             0
## CRC_05            1             0
## CRC_06            1             0</code></pre>
<p><strong>Set parameters and run main WGCNA function</strong></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="wgcna-visualizations.html#cb101-1" tabindex="-1"></a><span class="co">#parameters for WGCNA</span></span>
<span id="cb101-2"><a href="wgcna-visualizations.html#cb101-2" tabindex="-1"></a>power<span class="ot">=</span> <span class="dv">4</span></span>
<span id="cb101-3"><a href="wgcna-visualizations.html#cb101-3" tabindex="-1"></a>minModuleSize <span class="ot">=</span> <span class="dv">20</span></span>
<span id="cb101-4"><a href="wgcna-visualizations.html#cb101-4" tabindex="-1"></a>metadata_binary<span class="ot">=</span> design</span>
<span id="cb101-5"><a href="wgcna-visualizations.html#cb101-5" tabindex="-1"></a></span>
<span id="cb101-6"><a href="wgcna-visualizations.html#cb101-6" tabindex="-1"></a><span class="co">#Run WGCNA</span></span>
<span id="cb101-7"><a href="wgcna-visualizations.html#cb101-7" tabindex="-1"></a>net <span class="ot">=</span> <span class="fu">blockwiseModules</span>(datExpr, <span class="at">corType =</span> <span class="st">&quot;pearson&quot;</span>, <span class="at">maxBlockSize =</span> <span class="dv">5000</span>, </span>
<span id="cb101-8"><a href="wgcna-visualizations.html#cb101-8" tabindex="-1"></a>                       <span class="at">networkType =</span> <span class="st">&quot;signed&quot;</span>, <span class="at">power =</span> power, <span class="at">minModuleSize =</span>minModuleSize,</span>
<span id="cb101-9"><a href="wgcna-visualizations.html#cb101-9" tabindex="-1"></a>                       <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>, </span>
<span id="cb101-10"><a href="wgcna-visualizations.html#cb101-10" tabindex="-1"></a>                       <span class="at">numericLabels =</span> F, <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb101-11"><a href="wgcna-visualizations.html#cb101-11" tabindex="-1"></a>                       <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>, <span class="at">saveTOMFileBase =</span> <span class="st">&quot;TOM&quot;</span>)</span></code></pre></div>
<div id="module-phenotype-correlation-plot" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> <strong>Module-phenotype Correlation Plot</strong><a href="wgcna-visualizations.html#module-phenotype-correlation-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="wgcna-visualizations.html#cb102-1" tabindex="-1"></a><span class="co">#Module trait correlation (visualize which module associated with what phenotype)</span></span>
<span id="cb102-2"><a href="wgcna-visualizations.html#cb102-2" tabindex="-1"></a><span class="co">#We apply pearson correlation between metadata traits (conditions) and module eigen gene (1st principal component of module genes/metabolites) </span></span>
<span id="cb102-3"><a href="wgcna-visualizations.html#cb102-3" tabindex="-1"></a></span>
<span id="cb102-4"><a href="wgcna-visualizations.html#cb102-4" tabindex="-1"></a>plotheatmap<span class="ot">=</span><span class="cf">function</span>(datExpr, design, module_eigengenes){</span>
<span id="cb102-5"><a href="wgcna-visualizations.html#cb102-5" tabindex="-1"></a>traits<span class="ot">=</span> design <span class="sc">|&gt;</span>  <span class="fu">as.data.frame</span>()</span>
<span id="cb102-6"><a href="wgcna-visualizations.html#cb102-6" tabindex="-1"></a><span class="fu">head</span>(traits)</span>
<span id="cb102-7"><a href="wgcna-visualizations.html#cb102-7" tabindex="-1"></a><span class="co"># Define numbers of genes and samples</span></span>
<span id="cb102-8"><a href="wgcna-visualizations.html#cb102-8" tabindex="-1"></a>nSamples <span class="ot">&lt;-</span> <span class="fu">nrow</span>(datExpr)</span>
<span id="cb102-9"><a href="wgcna-visualizations.html#cb102-9" tabindex="-1"></a>nGenes <span class="ot">&lt;-</span> <span class="fu">ncol</span>(datExpr)</span>
<span id="cb102-10"><a href="wgcna-visualizations.html#cb102-10" tabindex="-1"></a>module.trait.corr <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">cor</span>(module_eigengenes, traits, <span class="at">use =</span> <span class="st">&#39;p&#39;</span>)</span>
<span id="cb102-11"><a href="wgcna-visualizations.html#cb102-11" tabindex="-1"></a>module.trait.corr.pvals <span class="ot">&lt;-</span> <span class="fu">corPvalueStudent</span>(module.trait.corr, nSamples)</span>
<span id="cb102-12"><a href="wgcna-visualizations.html#cb102-12" tabindex="-1"></a></span>
<span id="cb102-13"><a href="wgcna-visualizations.html#cb102-13" tabindex="-1"></a><span class="co">#module_trait heatmap of WGCNA package</span></span>
<span id="cb102-14"><a href="wgcna-visualizations.html#cb102-14" tabindex="-1"></a><span class="co"># correlations and their p-values</span></span>
<span id="cb102-15"><a href="wgcna-visualizations.html#cb102-15" tabindex="-1"></a>textMatrix <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">signif</span>(module.trait.corr, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>, <span class="fu">signif</span>(module.trait.corr.pvals, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, </span>
<span id="cb102-16"><a href="wgcna-visualizations.html#cb102-16" tabindex="-1"></a>                   <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb102-17"><a href="wgcna-visualizations.html#cb102-17" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">=</span> <span class="fu">dim</span>(module.trait.corr)</span>
<span id="cb102-18"><a href="wgcna-visualizations.html#cb102-18" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">6</span>))</span>
<span id="cb102-19"><a href="wgcna-visualizations.html#cb102-19" tabindex="-1"></a>color<span class="ot">=</span> <span class="fu">colorpanel</span>(<span class="dv">250</span>, <span class="st">&quot;#667F9C&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#FE1B1B&quot;</span>)</span>
<span id="cb102-20"><a href="wgcna-visualizations.html#cb102-20" tabindex="-1"></a></span>
<span id="cb102-21"><a href="wgcna-visualizations.html#cb102-21" tabindex="-1"></a><span class="co">#color= greenWhiteRed(50)</span></span>
<span id="cb102-22"><a href="wgcna-visualizations.html#cb102-22" tabindex="-1"></a><span class="co"># Display the correlation values within a heatmap plot</span></span>
<span id="cb102-23"><a href="wgcna-visualizations.html#cb102-23" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> module.trait.corr, <span class="at">xLabels =</span> <span class="fu">gsub</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">names</span>(traits)), </span>
<span id="cb102-24"><a href="wgcna-visualizations.html#cb102-24" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(module_eigengenes), </span>
<span id="cb102-25"><a href="wgcna-visualizations.html#cb102-25" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(module_eigengenes), <span class="at">colorLabels =</span> <span class="cn">FALSE</span>, <span class="at">colors =</span> color, </span>
<span id="cb102-26"><a href="wgcna-visualizations.html#cb102-26" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix, <span class="at">setStdMargins =</span> T, <span class="at">cex.text =</span> <span class="fl">0.8</span>,</span>
<span id="cb102-27"><a href="wgcna-visualizations.html#cb102-27" tabindex="-1"></a>               <span class="at">zlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>),<span class="at">xColorWidth =</span> <span class="dv">1</span> <span class="sc">*</span> <span class="fu">strheight</span>(<span class="st">&quot;M&quot;</span>),</span>
<span id="cb102-28"><a href="wgcna-visualizations.html#cb102-28" tabindex="-1"></a>               <span class="at">yColorWidth =</span> <span class="fl">1.5</span> <span class="sc">*</span> <span class="fu">strwidth</span>(<span class="st">&quot;M&quot;</span>),<span class="at">xColorOffset =</span> <span class="fu">strheight</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">/</span><span class="dv">6</span>, </span>
<span id="cb102-29"><a href="wgcna-visualizations.html#cb102-29" tabindex="-1"></a>               <span class="at">yColorOffset =</span> <span class="fu">strwidth</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">/</span><span class="dv">6</span>, <span class="at">font.lab.x =</span> <span class="dv">2</span>, <span class="at">cex.legendLabel =</span> <span class="dv">2</span>,</span>
<span id="cb102-30"><a href="wgcna-visualizations.html#cb102-30" tabindex="-1"></a>               <span class="at">font.lab.y =</span> <span class="dv">2</span>, <span class="at">xLabelsAngle =</span> <span class="dv">75</span>,</span>
<span id="cb102-31"><a href="wgcna-visualizations.html#cb102-31" tabindex="-1"></a>               <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module-Condition Relationship&quot;</span>), <span class="at">plotLegend=</span> <span class="cn">TRUE</span>)</span>
<span id="cb102-32"><a href="wgcna-visualizations.html#cb102-32" tabindex="-1"></a></span>
<span id="cb102-33"><a href="wgcna-visualizations.html#cb102-33" tabindex="-1"></a>}</span>
<span id="cb102-34"><a href="wgcna-visualizations.html#cb102-34" tabindex="-1"></a></span>
<span id="cb102-35"><a href="wgcna-visualizations.html#cb102-35" tabindex="-1"></a><span class="co">#Display</span></span>
<span id="cb102-36"><a href="wgcna-visualizations.html#cb102-36" tabindex="-1"></a><span class="fu">plotheatmap</span>(datExpr, design, module_eigengenes)</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-58-1.png" width="672" /></p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="wgcna-visualizations.html#cb103-1" tabindex="-1"></a><span class="co">#Save</span></span>
<span id="cb103-2"><a href="wgcna-visualizations.html#cb103-2" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;plots/heatmap_module_phenotype_cor.png&quot;</span>, <span class="at">width =</span> <span class="dv">4000</span>, <span class="at">height =</span> <span class="dv">4500</span>, <span class="at">res=</span> <span class="dv">600</span>) </span>
<span id="cb103-3"><a href="wgcna-visualizations.html#cb103-3" tabindex="-1"></a><span class="fu">plotheatmap</span>(datExpr, design, module_eigengenes)</span>
<span id="cb103-4"><a href="wgcna-visualizations.html#cb103-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="samples-dendrogram-plot" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> <strong>Samples Dendrogram Plot</strong><a href="wgcna-visualizations.html#samples-dendrogram-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="wgcna-visualizations.html#cb105-1" tabindex="-1"></a><span class="co">#Hierarchical clustering of samples, detect outlier samples,</span></span>
<span id="cb105-2"><a href="wgcna-visualizations.html#cb105-2" tabindex="-1"></a><span class="co">#and association of sample with certain trait</span></span>
<span id="cb105-3"><a href="wgcna-visualizations.html#cb105-3" tabindex="-1"></a>plotDendroAndphenotype<span class="ot">=</span> <span class="cf">function</span>(data, metadata){</span>
<span id="cb105-4"><a href="wgcna-visualizations.html#cb105-4" tabindex="-1"></a><span class="co">#Build adjacency matrix for samples</span></span>
<span id="cb105-5"><a href="wgcna-visualizations.html#cb105-5" tabindex="-1"></a>A <span class="ot">=</span> <span class="fu">adjacency</span>(data, <span class="at">type =</span> <span class="st">&quot;distance&quot;</span>)</span>
<span id="cb105-6"><a href="wgcna-visualizations.html#cb105-6" tabindex="-1"></a><span class="co"># this calculates the whole network connectivity</span></span>
<span id="cb105-7"><a href="wgcna-visualizations.html#cb105-7" tabindex="-1"></a>k <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(A, <span class="dv">2</span>, sum)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb105-8"><a href="wgcna-visualizations.html#cb105-8" tabindex="-1"></a><span class="co"># standardized connectivity</span></span>
<span id="cb105-9"><a href="wgcna-visualizations.html#cb105-9" tabindex="-1"></a>Z.k <span class="ot">=</span> <span class="fu">scale</span>(k)</span>
<span id="cb105-10"><a href="wgcna-visualizations.html#cb105-10" tabindex="-1"></a><span class="co"># Designate samples as outlying if their Z.k value is below the threshold</span></span>
<span id="cb105-11"><a href="wgcna-visualizations.html#cb105-11" tabindex="-1"></a>thresholdZ.k <span class="ot">=</span> <span class="sc">-</span><span class="dv">5</span>  <span class="co"># often -2.5</span></span>
<span id="cb105-12"><a href="wgcna-visualizations.html#cb105-12" tabindex="-1"></a><span class="co"># the color vector indicates outlyingness (red)</span></span>
<span id="cb105-13"><a href="wgcna-visualizations.html#cb105-13" tabindex="-1"></a>outlierColor <span class="ot">=</span> <span class="fu">ifelse</span>(Z.k <span class="sc">&lt;</span> thresholdZ.k, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;black&quot;</span>)</span>
<span id="cb105-14"><a href="wgcna-visualizations.html#cb105-14" tabindex="-1"></a><span class="co"># calculate the cluster tree using flahsClust or hclust</span></span>
<span id="cb105-15"><a href="wgcna-visualizations.html#cb105-15" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb105-16"><a href="wgcna-visualizations.html#cb105-16" tabindex="-1"></a><span class="co"># Convert traits to a color representation: where red indicates high</span></span>
<span id="cb105-17"><a href="wgcna-visualizations.html#cb105-17" tabindex="-1"></a><span class="co"># values</span></span>
<span id="cb105-18"><a href="wgcna-visualizations.html#cb105-18" tabindex="-1"></a>traitColors <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">numbers2colors</span>(<span class="fu">as.numeric</span>(metadata<span class="sc">$</span>cond_binary), <span class="at">signed =</span> <span class="cn">TRUE</span>))</span>
<span id="cb105-19"><a href="wgcna-visualizations.html#cb105-19" tabindex="-1"></a><span class="co">#dimnames(traitColors)[[2]] = &quot;Inflammation_lvl&quot;</span></span>
<span id="cb105-20"><a href="wgcna-visualizations.html#cb105-20" tabindex="-1"></a>datColors <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">outlier_Samples =</span> outlierColor, <span class="at">Condition=</span> traitColors)</span>
<span id="cb105-21"><a href="wgcna-visualizations.html#cb105-21" tabindex="-1"></a><span class="fu">colnames</span>(datColors)[<span class="dv">2</span>]<span class="ot">=</span> <span class="st">&quot;Conditions&quot;</span></span>
<span id="cb105-22"><a href="wgcna-visualizations.html#cb105-22" tabindex="-1"></a></span>
<span id="cb105-23"><a href="wgcna-visualizations.html#cb105-23" tabindex="-1"></a><span class="co"># Plot the sample dendrogram and the colors underneath.</span></span>
<span id="cb105-24"><a href="wgcna-visualizations.html#cb105-24" tabindex="-1"></a>WGCNA<span class="sc">::</span><span class="fu">plotDendroAndColors</span>(sampleTree, <span class="at">groupLabels =</span> <span class="fu">names</span>(datColors), <span class="at">colors =</span> datColors, <span class="at">cex.rowText =</span> <span class="dv">5</span>,  <span class="at">main =</span> <span class="st">&quot;Sample dendrogram and Homogeneity of samples heatmap&quot;</span>)</span>
<span id="cb105-25"><a href="wgcna-visualizations.html#cb105-25" tabindex="-1"></a></span>
<span id="cb105-26"><a href="wgcna-visualizations.html#cb105-26" tabindex="-1"></a>}</span>
<span id="cb105-27"><a href="wgcna-visualizations.html#cb105-27" tabindex="-1"></a></span>
<span id="cb105-28"><a href="wgcna-visualizations.html#cb105-28" tabindex="-1"></a><span class="co"># Display</span></span>
<span id="cb105-29"><a href="wgcna-visualizations.html#cb105-29" tabindex="-1"></a><span class="fu">plotDendroAndphenotype</span>(data,metadata)</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-59-1.png" width="672" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="wgcna-visualizations.html#cb106-1" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb106-2"><a href="wgcna-visualizations.html#cb106-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/WGCNA_dendrogram.png&quot;</span>), <span class="at">width =</span> <span class="dv">8000</span>, <span class="at">height =</span> <span class="dv">6000</span>, <span class="at">res=</span> <span class="dv">600</span>)</span>
<span id="cb106-3"><a href="wgcna-visualizations.html#cb106-3" tabindex="-1"></a><span class="fu">plotDendroAndphenotype</span>(data, metadata)</span>
<span id="cb106-4"><a href="wgcna-visualizations.html#cb106-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="cluster-dendrogram-plot" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> <strong>Cluster Dendrogram Plot</strong><a href="wgcna-visualizations.html#cluster-dendrogram-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="wgcna-visualizations.html#cb108-1" tabindex="-1"></a>plotDendro<span class="ot">=</span> <span class="cf">function</span>(){</span>
<span id="cb108-2"><a href="wgcna-visualizations.html#cb108-2" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]],  net<span class="sc">$</span>colors,</span>
<span id="cb108-3"><a href="wgcna-visualizations.html#cb108-3" tabindex="-1"></a>                    <span class="fu">paste0</span>(<span class="st">&quot;Merged</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="st">&quot;Modules&quot;</span>),</span>
<span id="cb108-4"><a href="wgcna-visualizations.html#cb108-4" tabindex="-1"></a>                    <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb108-5"><a href="wgcna-visualizations.html#cb108-5" tabindex="-1"></a>                    <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-6"><a href="wgcna-visualizations.html#cb108-6" tabindex="-1"></a>                    <span class="at">hang=</span> <span class="fl">0.03</span>,</span>
<span id="cb108-7"><a href="wgcna-visualizations.html#cb108-7" tabindex="-1"></a>                    <span class="at">cex.colorLabels =</span> <span class="fl">0.6</span>,</span>
<span id="cb108-8"><a href="wgcna-visualizations.html#cb108-8" tabindex="-1"></a>                    <span class="at">guideHang =</span> <span class="fl">0.05</span>)</span>
<span id="cb108-9"><a href="wgcna-visualizations.html#cb108-9" tabindex="-1"></a>}</span>
<span id="cb108-10"><a href="wgcna-visualizations.html#cb108-10" tabindex="-1"></a><span class="co"># Display</span></span>
<span id="cb108-11"><a href="wgcna-visualizations.html#cb108-11" tabindex="-1"></a><span class="fu">plotDendro</span>()</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-60-1.png" width="672" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="wgcna-visualizations.html#cb109-1" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb109-2"><a href="wgcna-visualizations.html#cb109-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/dendrogram_merged_modules.png&quot;</span>), <span class="at">width =</span> <span class="dv">2200</span>, <span class="at">height =</span> <span class="dv">2500</span>, <span class="at">res=</span> <span class="dv">600</span>)</span>
<span id="cb109-3"><a href="wgcna-visualizations.html#cb109-3" tabindex="-1"></a><span class="fu">plotDendro</span>()</span>
<span id="cb109-4"><a href="wgcna-visualizations.html#cb109-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="wgcna-visualizations.html#cb111-1" tabindex="-1"></a><span class="co">#for error, Error in .plotOrderedColorSubplot(order = order, colors = colors, rowLabels = rowLabels,  : </span></span>
<span id="cb111-2"><a href="wgcna-visualizations.html#cb111-2" tabindex="-1"></a><span class="co">#Length of colors vector not compatible with number of objects in &#39;order&#39;.</span></span>
<span id="cb111-3"><a href="wgcna-visualizations.html#cb111-3" tabindex="-1"></a><span class="co">#set good samples and good genes</span></span></code></pre></div>
</div>
<div id="tom-plot-heatmap-of-correlated-features-across-all-modules" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> <strong>TOM plot (Heatmap of Correlated Features Across All Modules)</strong><a href="wgcna-visualizations.html#tom-plot-heatmap-of-correlated-features-across-all-modules" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="wgcna-visualizations.html#cb112-1" tabindex="-1"></a>dissTOM<span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">TOMsimilarityFromExpr</span>(datExpr, <span class="at">power=</span> power) <span class="co">#datExpr samples in rows</span></span></code></pre></div>
<pre><code>## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.</code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="wgcna-visualizations.html#cb114-1" tabindex="-1"></a>dendro<span class="ot">=</span> net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb114-2"><a href="wgcna-visualizations.html#cb114-2" tabindex="-1"></a>moduleColorsAutomatic<span class="ot">=</span> net<span class="sc">$</span>colors</span>
<span id="cb114-3"><a href="wgcna-visualizations.html#cb114-3" tabindex="-1"></a></span>
<span id="cb114-4"><a href="wgcna-visualizations.html#cb114-4" tabindex="-1"></a>plotTOMheatmap <span class="ot">=</span> <span class="cf">function</span>() {</span>
<span id="cb114-5"><a href="wgcna-visualizations.html#cb114-5" tabindex="-1"></a>  <span class="co">#myheatcol = colorpanel(250,&#39;gold&#39;,&quot;orange&quot;,&#39;darkred&#39;)</span></span>
<span id="cb114-6"><a href="wgcna-visualizations.html#cb114-6" tabindex="-1"></a>  myheatcol <span class="ot">=</span> <span class="fu">colorpanel</span>(<span class="dv">250</span>,<span class="st">&#39;red&#39;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&#39;lemonchiffon&#39;</span>)</span>
<span id="cb114-7"><a href="wgcna-visualizations.html#cb114-7" tabindex="-1"></a>  <span class="co"># Transform dissTOM with a power to enhance visibility</span></span>
<span id="cb114-8"><a href="wgcna-visualizations.html#cb114-8" tabindex="-1"></a>  <span class="fu">TOMplot</span>(dissTOM, dendro, moduleColorsAutomatic,<span class="at">col=</span> myheatcol, </span>
<span id="cb114-9"><a href="wgcna-visualizations.html#cb114-9" tabindex="-1"></a>          <span class="at">main =</span> <span class="st">&quot;Module Heatmap Plot, All Features&quot;</span>)</span>
<span id="cb114-10"><a href="wgcna-visualizations.html#cb114-10" tabindex="-1"></a>}</span>
<span id="cb114-11"><a href="wgcna-visualizations.html#cb114-11" tabindex="-1"></a><span class="co"># Display</span></span>
<span id="cb114-12"><a href="wgcna-visualizations.html#cb114-12" tabindex="-1"></a><span class="fu">plotTOMheatmap</span>()</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="wgcna-visualizations.html#cb115-1" tabindex="-1"></a><span class="co"># save</span></span>
<span id="cb115-2"><a href="wgcna-visualizations.html#cb115-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/TOM_PLOT_module_heatmap_all.png&quot;</span>), <span class="at">width =</span> <span class="dv">800</span>, <span class="at">height =</span> <span class="dv">600</span>)</span>
<span id="cb115-3"><a href="wgcna-visualizations.html#cb115-3" tabindex="-1"></a><span class="fu">plotTOMheatmap</span>()</span>
<span id="cb115-4"><a href="wgcna-visualizations.html#cb115-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="mds-plot" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> <strong>MDS plot</strong><a href="wgcna-visualizations.html#mds-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="wgcna-visualizations.html#cb117-1" tabindex="-1"></a>dissTOM<span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">TOMsimilarityFromExpr</span>(datExpr, <span class="at">power=</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="wgcna-visualizations.html#cb119-1" tabindex="-1"></a>cmd1<span class="ot">=</span><span class="fu">cmdscale</span>(<span class="fu">as.dist</span>(dissTOM),<span class="dv">2</span>)</span>
<span id="cb119-2"><a href="wgcna-visualizations.html#cb119-2" tabindex="-1"></a><span class="co"># Display</span></span>
<span id="cb119-3"><a href="wgcna-visualizations.html#cb119-3" tabindex="-1"></a><span class="fu">plot</span>(cmd1,<span class="at">col=</span>moduleColorsAutomatic,<span class="at">main=</span><span class="st">&quot;MDS plot&quot;</span>,</span>
<span id="cb119-4"><a href="wgcna-visualizations.html#cb119-4" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Scaling Dimension 1&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scaling Dimension 2&quot;</span>)</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-62-1.png" width="672" /></p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="wgcna-visualizations.html#cb120-1" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb120-2"><a href="wgcna-visualizations.html#cb120-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/MDS_plot.png&quot;</span>),  <span class="at">width =</span> <span class="dv">2800</span>, <span class="at">height =</span> <span class="dv">3300</span>, <span class="at">res=</span> <span class="dv">600</span>) </span>
<span id="cb120-3"><a href="wgcna-visualizations.html#cb120-3" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb120-4"><a href="wgcna-visualizations.html#cb120-4" tabindex="-1"></a><span class="fu">plot</span>(cmd1,<span class="at">col=</span>moduleColorsAutomatic,<span class="at">main=</span><span class="st">&quot;MDS plot&quot;</span>,</span>
<span id="cb120-5"><a href="wgcna-visualizations.html#cb120-5" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Scaling Dimension 1&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scaling Dimension 2&quot;</span>)</span>
<span id="cb120-6"><a href="wgcna-visualizations.html#cb120-6" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="tom-plot-for-selected-features" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> <strong>TOM plot for Selected Features</strong><a href="wgcna-visualizations.html#tom-plot-for-selected-features" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="wgcna-visualizations.html#cb122-1" tabindex="-1"></a><span class="co"># Get Differentially expressed features</span></span>
<span id="cb122-2"><a href="wgcna-visualizations.html#cb122-2" tabindex="-1"></a>genes<span class="ot">=</span> de<span class="sc">$</span>X</span>
<span id="cb122-3"><a href="wgcna-visualizations.html#cb122-3" tabindex="-1"></a><span class="co"># subset data to have only selected proteins</span></span>
<span id="cb122-4"><a href="wgcna-visualizations.html#cb122-4" tabindex="-1"></a>datExpr_subset<span class="ot">=</span> datExpr[,<span class="fu">colnames</span>(datExpr) <span class="sc">%in%</span> genes]</span>
<span id="cb122-5"><a href="wgcna-visualizations.html#cb122-5" tabindex="-1"></a>dissTOM_subset<span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">TOMsimilarityFromExpr</span>(datExpr_subset, <span class="at">power=</span> power)</span></code></pre></div>
<pre><code>## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.</code></pre>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="wgcna-visualizations.html#cb124-1" tabindex="-1"></a>dendro_subset <span class="ot">=</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(dissTOM_subset), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb124-2"><a href="wgcna-visualizations.html#cb124-2" tabindex="-1"></a>module.gene.assign<span class="ot">=</span> net<span class="sc">$</span>colors</span>
<span id="cb124-3"><a href="wgcna-visualizations.html#cb124-3" tabindex="-1"></a>moduleColors_subset<span class="ot">=</span> module.gene.assign[<span class="fu">names</span>(module.gene.assign)<span class="sc">%in%</span> genes]</span>
<span id="cb124-4"><a href="wgcna-visualizations.html#cb124-4" tabindex="-1"></a></span>
<span id="cb124-5"><a href="wgcna-visualizations.html#cb124-5" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/module_heatmap_TOM_PLOT_selected.png&quot;</span>), <span class="at">width =</span> <span class="dv">2800</span>, <span class="at">height =</span> <span class="dv">3300</span>, <span class="at">res=</span> <span class="dv">600</span>)</span>
<span id="cb124-6"><a href="wgcna-visualizations.html#cb124-6" tabindex="-1"></a><span class="co">#myheatcol = colorpanel(250,&#39;gold&#39;,&quot;orange&quot;,&#39;darkred&#39;)</span></span>
<span id="cb124-7"><a href="wgcna-visualizations.html#cb124-7" tabindex="-1"></a>myheatcol <span class="ot">=</span> <span class="fu">colorpanel</span>(<span class="dv">250</span>,<span class="st">&#39;red&#39;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&#39;lemonchiffon&#39;</span>)</span>
<span id="cb124-8"><a href="wgcna-visualizations.html#cb124-8" tabindex="-1"></a><span class="co"># Transform dissTOM with a power to enhance visibility</span></span>
<span id="cb124-9"><a href="wgcna-visualizations.html#cb124-9" tabindex="-1"></a><span class="fu">TOMplot</span>(dissTOM_subset, </span>
<span id="cb124-10"><a href="wgcna-visualizations.html#cb124-10" tabindex="-1"></a>        dendro_subset, </span>
<span id="cb124-11"><a href="wgcna-visualizations.html#cb124-11" tabindex="-1"></a>        moduleColors_subset,<span class="at">col=</span> myheatcol,</span>
<span id="cb124-12"><a href="wgcna-visualizations.html#cb124-12" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">&quot;Module Heatmap Plot&quot;</span>)</span>
<span id="cb124-13"><a href="wgcna-visualizations.html#cb124-13" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="wgcna-visualizations.html#cb126-1" tabindex="-1"></a><span class="co"># Intramodular analysis: identifying genes with high Gene Significance (GS), i.e., strong association with the phenotype, </span></span>
<span id="cb126-2"><a href="wgcna-visualizations.html#cb126-2" tabindex="-1"></a><span class="co">#and high Module Membership (MM), i.e., strong correlation with the module eigengene (first principal component).</span></span>
<span id="cb126-3"><a href="wgcna-visualizations.html#cb126-3" tabindex="-1"></a>datKME <span class="ot">=</span> <span class="fu">signedKME</span>(datExpr, module_eigengenes)</span>
<span id="cb126-4"><a href="wgcna-visualizations.html#cb126-4" tabindex="-1"></a>GS.lvl<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/gene.trait.corr.csv&quot;</span>)) <span class="sc">|&gt;</span>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;X&quot;</span>)</span>
<span id="cb126-5"><a href="wgcna-visualizations.html#cb126-5" tabindex="-1"></a>modules_df<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/modules.csv&quot;</span>))</span>
<span id="cb126-6"><a href="wgcna-visualizations.html#cb126-6" tabindex="-1"></a>module.assign<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;output/gene_module_assignment.csv&quot;</span>)) <span class="sc">|&gt;</span>  <span class="fu">deframe</span>() <span class="sc">|&gt;</span>  <span class="fu">unlist</span>()</span>
<span id="cb126-7"><a href="wgcna-visualizations.html#cb126-7" tabindex="-1"></a><span class="fu">dim</span>(datKME)</span></code></pre></div>
<pre><code>## [1] 263   4</code></pre>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="wgcna-visualizations.html#cb128-1" tabindex="-1"></a><span class="fu">dim</span>(GS.lvl)</span></code></pre></div>
<pre><code>## [1] 263   2</code></pre>
</div>
<div id="module-membership-vs-gene-significance-scatter-plot" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> <strong>Module Membership Vs Gene Significance Scatter plot</strong><a href="wgcna-visualizations.html#module-membership-vs-gene-significance-scatter-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="wgcna-visualizations.html#cb130-1" tabindex="-1"></a><span class="co"># Define plotting function</span></span>
<span id="cb130-2"><a href="wgcna-visualizations.html#cb130-2" tabindex="-1"></a>plot_MM_vs_GS <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb130-3"><a href="wgcna-visualizations.html#cb130-3" tabindex="-1"></a>  colorOfColumn <span class="ot">&lt;-</span> <span class="fu">substring</span>(<span class="fu">names</span>(datKME), <span class="dv">4</span>)</span>
<span id="cb130-4"><a href="wgcna-visualizations.html#cb130-4" tabindex="-1"></a>  selectModules <span class="ot">&lt;-</span> <span class="fu">colnames</span>(modules_df)</span>
<span id="cb130-5"><a href="wgcna-visualizations.html#cb130-5" tabindex="-1"></a>  </span>
<span id="cb130-6"><a href="wgcna-visualizations.html#cb130-6" tabindex="-1"></a>  <span class="co"># Set layout: 2 rows, N/2 columns</span></span>
<span id="cb130-7"><a href="wgcna-visualizations.html#cb130-7" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>) <span class="sc">+</span> <span class="fl">0.1</span>)</span>
<span id="cb130-8"><a href="wgcna-visualizations.html#cb130-8" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="fu">ceiling</span>(<span class="fu">length</span>(selectModules) <span class="sc">/</span> <span class="dv">2</span>)))</span>
<span id="cb130-9"><a href="wgcna-visualizations.html#cb130-9" tabindex="-1"></a>  </span>
<span id="cb130-10"><a href="wgcna-visualizations.html#cb130-10" tabindex="-1"></a>  <span class="cf">for</span> (module <span class="cf">in</span> selectModules) {</span>
<span id="cb130-11"><a href="wgcna-visualizations.html#cb130-11" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span>(module <span class="sc">%in%</span> module.assign)) {</span>
<span id="cb130-12"><a href="wgcna-visualizations.html#cb130-12" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;Skipping module&quot;</span>, module, <span class="st">&quot;- not found in module assignment&quot;</span>))</span>
<span id="cb130-13"><a href="wgcna-visualizations.html#cb130-13" tabindex="-1"></a>      <span class="cf">next</span></span>
<span id="cb130-14"><a href="wgcna-visualizations.html#cb130-14" tabindex="-1"></a>    }</span>
<span id="cb130-15"><a href="wgcna-visualizations.html#cb130-15" tabindex="-1"></a></span>
<span id="cb130-16"><a href="wgcna-visualizations.html#cb130-16" tabindex="-1"></a>    column <span class="ot">&lt;-</span> <span class="fu">match</span>(module, colorOfColumn)</span>
<span id="cb130-17"><a href="wgcna-visualizations.html#cb130-17" tabindex="-1"></a>    restModule <span class="ot">&lt;-</span> moduleColorsAutomatic <span class="sc">==</span> module</span>
<span id="cb130-18"><a href="wgcna-visualizations.html#cb130-18" tabindex="-1"></a></span>
<span id="cb130-19"><a href="wgcna-visualizations.html#cb130-19" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(restModule) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb130-20"><a href="wgcna-visualizations.html#cb130-20" tabindex="-1"></a>      <span class="fu">verboseScatterplot</span>(</span>
<span id="cb130-21"><a href="wgcna-visualizations.html#cb130-21" tabindex="-1"></a>        datKME[restModule, column],</span>
<span id="cb130-22"><a href="wgcna-visualizations.html#cb130-22" tabindex="-1"></a>        GS.lvl[restModule, <span class="dv">1</span>],</span>
<span id="cb130-23"><a href="wgcna-visualizations.html#cb130-23" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="fu">paste</span>(<span class="st">&quot;Module Membership</span><span class="sc">\n</span><span class="st">&quot;</span>, module, <span class="st">&quot;module&quot;</span>),</span>
<span id="cb130-24"><a href="wgcna-visualizations.html#cb130-24" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">&quot;Feature Significance&quot;</span>,</span>
<span id="cb130-25"><a href="wgcna-visualizations.html#cb130-25" tabindex="-1"></a>        <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;kME:&quot;</span>, module, <span class="st">&quot;vs Feature Sig.&quot;</span>),</span>
<span id="cb130-26"><a href="wgcna-visualizations.html#cb130-26" tabindex="-1"></a>        <span class="at">col =</span> module</span>
<span id="cb130-27"><a href="wgcna-visualizations.html#cb130-27" tabindex="-1"></a>      )</span>
<span id="cb130-28"><a href="wgcna-visualizations.html#cb130-28" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb130-29"><a href="wgcna-visualizations.html#cb130-29" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;No matching entries for module:&quot;</span>, module))</span>
<span id="cb130-30"><a href="wgcna-visualizations.html#cb130-30" tabindex="-1"></a>    }</span>
<span id="cb130-31"><a href="wgcna-visualizations.html#cb130-31" tabindex="-1"></a>  }</span>
<span id="cb130-32"><a href="wgcna-visualizations.html#cb130-32" tabindex="-1"></a>}</span>
<span id="cb130-33"><a href="wgcna-visualizations.html#cb130-33" tabindex="-1"></a></span>
<span id="cb130-34"><a href="wgcna-visualizations.html#cb130-34" tabindex="-1"></a><span class="co">#Display </span></span>
<span id="cb130-35"><a href="wgcna-visualizations.html#cb130-35" tabindex="-1"></a><span class="fu">plot_MM_vs_GS</span>()</span></code></pre></div>
<p><img src="mybook_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="wgcna-visualizations.html#cb131-1" tabindex="-1"></a><span class="co">#Save</span></span>
<span id="cb131-2"><a href="wgcna-visualizations.html#cb131-2" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">paste0</span>(mypath,<span class="st">&quot;plots/mm_vs_sig.png&quot;</span>), <span class="at">width =</span> <span class="dv">800</span>, <span class="at">height =</span> <span class="dv">700</span>)</span>
<span id="cb131-3"><a href="wgcna-visualizations.html#cb131-3" tabindex="-1"></a><span class="fu">plot_MM_vs_GS</span>()</span>
<span id="cb131-4"><a href="wgcna-visualizations.html#cb131-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
<pre><code>## R version 4.4.1 (2024-06-14 ucrt)
## Platform: x86_64-w64-mingw32/x64
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## 
## locale:
## [1] LC_COLLATE=English_United States.utf8 
## [2] LC_CTYPE=English_United States.utf8   
## [3] LC_MONETARY=English_United States.utf8
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.utf8    
## 
## time zone: Africa/Cairo
## tzcode source: internal
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] gplots_3.1.3.1        gridExtra_2.3         flashClust_1.01-2    
##  [4] ggdendro_0.2.0        ape_5.8               pROC_1.18.5          
##  [7] gtools_3.9.5          WGCNA_1.72-5          fastcluster_1.2.6    
## [10] dynamicTreeCut_1.63-1 ggrepel_0.9.6         viridis_0.6.5        
## [13] fields_16.2           viridisLite_0.4.2     spam_2.10-0          
## [16] biomaRt_2.61.2        ComplexHeatmap_2.21.0 circlize_0.4.16      
## [19] RColorBrewer_1.1-3    memoise_2.0.1         caret_6.0-94         
## [22] lattice_0.22-6        pls_2.8-3             Rserve_1.8-13        
## [25] MetaboAnalystR_3.2.0  cowplot_1.1.3         DT_0.33              
## [28] openxlsx_4.2.6.1      lubridate_1.9.3       forcats_1.0.0        
## [31] stringr_1.5.1         purrr_1.0.2           readr_2.1.5          
## [34] tidyr_1.3.1           ggplot2_3.5.1         tidyverse_2.0.0      
## [37] dplyr_1.1.4           plyr_1.8.9            tibble_3.2.1         
## 
## loaded via a namespace (and not attached):
##   [1] matrixStats_1.3.0       bitops_1.0-7            httr_1.4.7             
##   [4] doParallel_1.0.17       tools_4.4.1             backports_1.5.0        
##   [7] R6_2.5.1                lazyeval_0.2.2          GetoptLong_1.0.5       
##  [10] withr_3.0.0             prettyunits_1.2.0       preprocessCore_1.67.0  
##  [13] cli_3.6.3               Biobase_2.64.0          textshaping_0.4.0      
##  [16] Cairo_1.6-2             labeling_0.4.3          sass_0.4.9             
##  [19] proxy_0.4-27            systemfonts_1.2.3       foreign_0.8-86         
##  [22] siggenes_1.79.0         parallelly_1.38.0       scrime_1.3.5           
##  [25] maps_3.4.2              limma_3.61.5            rstudioapi_0.16.0      
##  [28] impute_1.79.0           RSQLite_2.3.7           generics_0.1.3         
##  [31] shape_1.4.6.1           RApiSerialize_0.1.3     crmn_0.0.21            
##  [34] crosstalk_1.2.1         zip_2.3.1               GO.db_3.19.1           
##  [37] Matrix_1.7-0            S4Vectors_0.42.1        lifecycle_1.0.4        
##  [40] yaml_2.3.10             edgeR_4.3.5             recipes_1.1.0          
##  [43] BiocFileCache_2.13.0    blob_1.2.4              crayon_1.5.3           
##  [46] KEGGREST_1.45.1         magick_2.8.4            pillar_1.11.0          
##  [49] knitr_1.48              fgsea_1.31.0            rjson_0.2.21           
##  [52] future.apply_1.11.2     codetools_0.2-20        fastmatch_1.1-4        
##  [55] glue_1.7.0              pcaMethods_1.97.0       data.table_1.15.4      
##  [58] vctrs_0.6.5             png_0.1-8               gtable_0.3.5           
##  [61] cachem_1.1.0            gower_1.0.1             xfun_0.46              
##  [64] prodlim_2024.06.25      survival_3.6-4          timeDate_4032.109      
##  [67] iterators_1.0.14        hardhat_1.4.0           lava_1.8.0             
##  [70] statmod_1.5.0           ipred_0.9-15            nlme_3.1-164           
##  [73] bit64_4.0.5             progress_1.2.3          filelock_1.0.3         
##  [76] GenomeInfoDb_1.41.1     bslib_0.8.0             KernSmooth_2.23-24     
##  [79] rpart_4.1.23            colorspace_2.1-1        BiocGenerics_0.52.0    
##  [82] DBI_1.2.3               Hmisc_5.1-3             nnet_7.3-19            
##  [85] tidyselect_1.2.1        bit_4.0.5               compiler_4.4.1         
##  [88] curl_5.2.1              httr2_1.0.2             htmlTable_2.4.3        
##  [91] xml2_1.3.6              plotly_4.10.4           stringfish_0.16.0      
##  [94] bookdown_0.40           checkmate_2.3.1         scales_1.3.0           
##  [97] caTools_1.18.2          rappdirs_0.3.3          digest_0.6.36          
## [100] rmarkdown_2.27          XVector_0.44.0          htmltools_0.5.8.1      
## [103] pkgconfig_2.0.3         base64enc_0.1-3         highr_0.11             
## [106] dbplyr_2.5.0            fastmap_1.2.0           rlang_1.1.4            
## [109] GlobalOptions_0.1.2     htmlwidgets_1.6.4       UCSC.utils_1.1.0       
## [112] farver_2.1.2            jquerylib_0.1.4         jsonlite_1.8.8         
## [115] BiocParallel_1.39.0     ModelMetrics_1.2.2.2    magrittr_2.0.3         
## [118] Formula_1.2-5           GenomeInfoDbData_1.2.12 dotCall64_1.1-1        
## [121] munsell_0.5.1           Rcpp_1.0.13             stringi_1.8.4          
## [124] zlibbioc_1.50.0         MASS_7.3-60.2           parallel_4.4.1         
## [127] listenv_0.9.1           Biostrings_2.72.1       splines_4.4.1          
## [130] multtest_2.61.0         hms_1.1.3               locfit_1.5-9.10        
## [133] igraph_2.0.3            reshape2_1.4.4          stats4_4.4.1           
## [136] evaluate_0.24.0         RcppParallel_5.1.8      tzdb_0.4.0             
## [139] foreach_1.5.2           qs_0.26.3               future_1.33.2          
## [142] clue_0.3-65             e1071_1.7-14            glasso_1.11            
## [145] class_7.3-22            ragg_1.3.2              AnnotationDbi_1.67.0   
## [148] ellipse_0.5.0           IRanges_2.38.1          cluster_2.1.6          
## [151] timechange_0.3.0        globals_0.16.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="wgcna-pipeline.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-eigengene-differentiation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/Menna-Arafat/CRC_project/blob/master/6_WGCNA_visualizations.rmd",
"text": null
},
"download": ["mybook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
